---
title: "RNAseq_QC"
author: "Emma M Pfortmiller"
date: "2025-09-26"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
  
}
```

#Load Libraries
```{r Necessary Libraries, message=FALSE, warning=FALSE, include=FALSE}
library(tidyverse)
library(Biobase)
library(limma)
library(edgeR)
library(edgebundleR)
library(scales)
library(biomaRt)
library(ggrepel)
library(ggfortify)
library(corrplot)
library(readr)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(ComplexHeatmap)
library(circlize)
library(grid)
library(reshape2)
library(ggVennDiagram)
library(UpSetR)
library(ggpubr)
library(VennDiagram)
library(VennDetail)
library(BiocParallel)
library(RUVSeq)
library(SummarizedExperiment)
library(ggsignif)
library(purrr)
library(rstatix)
library(ggrastr)
library(tibble)
library(grid)
library(eulerr)
library(patchwork)
library(car)
library(gprofiler2)
library(forcats)
library(broom)
library(magick)
```

#Define plot theme
```{r Custom Theme}
# Define the custom theme
# plot_theme_custom <- function() {
#   theme_minimal() +
#     theme(
#       #line for x and y axis
#       axis.line = element_line(linewidth = 1,
#                                color = "black"),
# 
#       #axis ticks only on x and y, length standard
#       axis.ticks.x = element_line(color = "black",
#                                 linewidth = 1),
#       axis.ticks.y = element_line(color = "black",
#                                 linewidth = 1),
#       axis.ticks.length = unit(0.05, "in"),
# 
#       #text and font
#       axis.text = element_text(color = "black",
#                                family = "Arial",
#                                size = 8),
#       axis.title = element_text(color = "black",
#                                 family = "Arial",
#                                 size = 10),
#       legend.text = element_text(color = "black",
#                                  family = "Arial",
#                                  size = 8),
#       legend.title = element_text(color = "black",
#                                   family = "Arial",
#                                   size = 10),
#       plot.title = element_text(color = "black",
#                                 family = "Arial",
#                                 size = 12),
# 
#       #blank background and border
#       panel.background = element_blank(),
#       panel.border = element_blank(),
# 
#       #gridlines for alignment
#       panel.grid.major = element_line(color = "grey80", linewidth = 0.5),  #grey major grid for align in illus
#       panel.grid.minor = element_line(color = "grey90", linewidth = 0.5) #grey minor grid for align in illus
#     )
# }

# saveRDS(plot_theme_custom, "data/plot_theme_custom.RDS")

theme_custom <- readRDS("data/plot_theme_custom.RDS")

```

#Define saving plots as pdfs
```{r pdf saving function}

save_plot <- function(plot, filename, 
                      folder = ".", 
                      width = 8, 
                      height = 6, 
                      units = "in", 
                      dpi = 300, 
                      add_date = TRUE) {
  if (missing(filename)) stop("Please provide a filename (without extension) for the plot.")

  date_str <- if (add_date) paste0("_", format(Sys.Date(), "%y%m%d")) else ""
  pdf_file <- file.path(folder, paste0(filename, date_str, ".pdf"))
  png_file <- file.path(folder, paste0(filename, date_str, ".png"))
  ggsave(filename = pdf_file, plot = plot, device = cairo_pdf, width = width, height = height, units = units, bg = "transparent")
  ggsave(filename = png_file, plot = plot, device = "png", width = width, height = height, units = units, dpi = dpi, bg = "transparent")
  message("Saved plot as Cairo PDF: ", pdf_file)
  message("Saved plot as PNG: ", png_file)
}

output_folder <- "C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/Stressor Project/Full Set RNAseq/plots"


#save plot function created
#to use: just define the plot name, filename_base, width, height
```

#Factors and Colors
```{r Theme Colors and Factors}
#each of these to be used after pairwise comparison save individual

####STIMULI####
# stim_list <- list(
#   "Tunicamycin" = "TUN",
#   "Thapsigargin" = "THA",
#   "Doxorubicin" = "DOX",
#   "Nutlin-3" = "NUTL",
#   "Lipopolysaccharides" = "LPS",
#   "Tumor Necrosis Factor alpha" = "TNFa",
#   "Bisphenol A" = "BPA",
#   "Perfluorooctanoic Acid" = "PFOA"
# )
# stimuli_vec <- unlist(lapply(names(stim_list), function(cat) {
#   setNames(rep(cat, length(stim_list[[cat]])), stim_list[[cat]])
# }))
# saveRDS(stimuli_vec, "data/theme/stimuli_fullname_vector.RDS")
stim_vec <- readRDS("data/theme/stimuli_fullname_vector.RDS")

####RESPONSE CATEGORY####
# resp_list <- list(
#   UPR = c("TUN", "THA"),
#   DDR = c("DOX", "NUTL"),
#   IMR = c("LPS", "TNFa"),
#   MMR = c("BPA", "PFOA")
# )
# response_vec <- unlist(lapply(names(resp_list), function(cat) {
#   setNames(rep(cat, length(resp_list[[cat]])), resp_list[[cat]])
# }))
# saveRDS(response_vec, "data/theme/response_categories_vector.RDS")
response_vec <- readRDS("data/theme/response_categories_vector.RDS")

####SPECIES####
# spec_list <- list(
#   Human = "H",
#   Chimp = "C"
# )
# species_vec <- unlist(lapply(names(spec_list), function(cat) {
#   setNames(rep(cat, length(spec_list[[cat]])), spec_list[[cat]])
# }))
# saveRDS(species_vec, "data/theme/species_vector.RDS")
species_vec <- readRDS("data/theme/species_vector.RDS")

####TIME####
# time_list <- list(
#   "2hr" = "2",
#   "24hr" = "24"
# )
# time_vec <- unlist(lapply(names(time_list), function(cat) {
#   setNames(rep(cat, length(time_list[[cat]])), time_list[[cat]])
# }))
# saveRDS(time_vec, "data/theme/time_vector.RDS")
time_vec <- readRDS("data/theme/time_vector.RDS")

####INDIVIDUAL####
# ind_list <- list(
#   Ind1 = "1",
#   Ind2 = "2",
#   Ind3 = "3",
#   Ind4 = "4",
#   Ind5 = "5",
#   Ind6 = "6",
#   Ind7 = "7",
#   Ind8 = "8",
#   Ind9 = "9",
#   Ind10 = "10",
#   Ind11 = "11",
#   Ind12 = "12",
#   Ind13 = "13",
#   Ind14 = "14"
# )
# ind_vec <- unlist(lapply(names(ind_list), function (cat) {
#   setNames(rep(cat, length(ind_list[[cat]])), ind_list[[cat]])
# }))
# saveRDS(ind_vec, "data/theme/individual_vector.RDS")
ind_vec <- readRDS("data/theme/individual_vector.RDS")

```


#Datasets Creation
```{r Import Counts Tables}
#I have one counts table for chimpanzee and another for human

#read in chimpanzee counts
# c_raw_counts <- read_csv("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/Stressor Project/Full Set RNAseq/featurecounts/c_samples_counts_fin.csv")

# saveRDS(c_raw_counts, "data/counts/c_raw_counts.RDS")

c_raw_counts <- readRDS("data/counts/c_raw_counts.RDS")

#read in human counts table in the same manner

# h_raw_counts <- read_csv("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/Stressor Project/Full Set RNAseq/featurecounts/h_samples_counts_fin.csv")

# saveRDS(h_raw_counts, "data/counts/h_raw_counts.RDS")

h_raw_counts <- readRDS("data/counts/h_raw_counts.RDS")

```

#Metadata
```{r Read in Metadata Sheet}
#I have a metadata sheet so I can associate the column names with their sample ID

# metadata <- read_csv("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/Stressor Project/Full Set RNAseq/featurecounts/dissrt_metadata.csv",
#     col_types = cols(Conc = col_number()))

#save metadata sheet

# saveRDS(metadata, "data/counts/dissrt_metadata.RDS")

metadata <- readRDS("data/counts/dissrt_metadata.RDS")

#subset my metadata 
human_metadata <- metadata[metadata$Species == "H", ]
chimp_metadata <- metadata[metadata$Species == "C", ]

```

```{r Change Column Names}
#I want to modify the column names for each dataset to match their sample id and treatment instead of library id

#first make sure that Geneid changes to Ensembl_ID
# rename_with_metadata <- function(counts, metadata_subset) {
# #
#   # 1. Rename first column to Ensembl_ID if not already
#   colnames(counts)[1] <- "Ensembl_ID"
# 
#   # 2. Columns to rename (all except first)
#   cols_to_rename <- colnames(counts)[-1]
# 
#   # 3. Remove ".sorted.bam" from column names
#   library_names <- gsub("\\.sorted\\.bam$", "", cols_to_rename)
# 
#   # 4. Match library names to Sample_name in metadata
#   new_names <- metadata_subset$Sample_Tx[match(library_names, metadata_subset$Sample_ID)]
# 
#   # 5. Warn if any columns didnâ€™t match
#   unmatched <- cols_to_rename[is.na(new_names)]
#   if (length(unmatched) > 0) {
#     warning("The following library names did not match metadata: ", paste(unmatched, collapse = ", "))
#   }
# 
#   # 6. Replace column names (keep first column as Ensembl_ID)
#   colnames(counts) <- c("Ensembl_ID", ifelse(is.na(new_names), cols_to_rename, new_names))
# 
#   return(counts)
# }

# Subset metadata by species
# human_metadata <- metadata[metadata$Species == "H", ]
# chimp_metadata <- metadata[metadata$Species == "C", ]

# Apply to counts tables
# h_counts <- rename_with_metadata(h_raw_counts, human_metadata)
# c_counts <- rename_with_metadata(c_raw_counts, chimp_metadata)

# save new counts files that have correctly named columns
# saveRDS(h_counts, "data/counts/h_counts.RDS")
# saveRDS(c_counts, "data/counts/c_counts.RDS")

#kept all of this for posterity but don't need to run again

#read in final files
h_counts <- readRDS("data/counts/h_counts.RDS")
c_counts <- readRDS("data/counts/c_counts.RDS")

```

#Combine my datasets
```{r Join my Human and Chimp datasets together}
#make sure to sort them so that all of the genes match up
# 
# h_counts <- h_counts[order(h_counts$Ensembl_ID), ]
# c_counts <- c_counts[order(c_counts$Ensembl_ID), ]
# 
# hc_counts <- left_join(c_counts, h_counts, by = "Ensembl_ID")

#now save my combined counts table with both species
# saveRDS(hc_counts, "data/counts/combined_counts_table_h_c.RDS")

# hc_counts <- readRDS("data/counts/combined_counts_table_h_c.RDS")

#now I have a combined dataframe with the treatment names included

####Replace hc_counts with one made below that has Entrez_ID instead of Ensembl_ID

hc_counts <- readRDS("data/counts/hc_counts_entrez.RDS")

```

#QC Mapping/fC
```{r Mapping and Counts Stats}
#read in my excel file with all of this information
# map_align_stats <- read_excel("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/Stressor Project/Full Set RNAseq/featurecounts/Mapping_Alignment_Stats_Dissrt_EMP_250930.xlsx",
#     sheet = "Map_Align_all", col_types = c("skip",
#         "text", "numeric", "text" , "text", "text",
#         "text", "numeric", "text", "text",
#         "text", "text", "text", "numeric",
#         "text", "numeric", "numeric", "numeric",
#         "numeric", "numeric", "numeric",
#         "numeric", "numeric", "numeric",
#         "numeric", "numeric", "numeric",
#         "numeric"))
# View(map_align_stats)

#save as a csv for later
# saveRDS(map_align_stats, "data/counts/mapping_alignment_stats_sheet.RDS")
# write.csv(map_align_stats, "data/counts/mapping_alignment_stats_sheet.csv")

map_align_stats <- readRDS("data/counts/mapping_alignment_stats_sheet.RDS")

#now I want to do some QC plots to highlight this information

Individual <- as.character(map_align_stats$Ind)
Species <- as.character(map_align_stats$Species)
Time <- as.character(map_align_stats$Time)
Stimulus <- as.character(map_align_stats$Stimulus)

ind_col <- setNames(rainbow(length(unique(Individual))), unique(Individual))


time_col <- c("2" = "#392344",
              "24" = "#457291")

species_col <- c("C" = "#E56473",
                 "H" = "#EE4902")

####Reads by Sample####
reads_by_sample <- c(
  "TUN" = "#E41A1C",
  "THA" = "#377EB8", 
  "DOX" = "#4DAF4A", 
  "NUTL" = "#984EA3", 
  "DMSO" = "#999999", 
  "LPS" = "#FFFF33", 
  "TNFa" = "#FC8D62", 
  "H2O" = "#777777", 
  "BPA" = "#F781BF", 
  "PFOA" = "#66C2A5", 
  "EtOH" = "#AAAAAA"
)

map_align_stats$Stimulus <- factor(
  map_align_stats$Stimulus,
  levels = c("TUN", "THA", "DOX", "NUTL", "DMSO", 
             "LPS", "TNFa", "H2O", "BPA", "PFOA", "EtOH")
)

map_align_stats$Ind <- factor(
  map_align_stats$Ind,
  levels = c("Ind1", "Ind2", "Ind3", "Ind4", "Ind5", 
             "Ind6", "Ind7", "Ind8", "Ind9", "Ind10", 
             "Ind11", "Ind12", "Ind13", "Ind14")
)

#Total Reads per Sample
total_reads_sample_plot <- map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y = Total_Reads, 
                 fill = Stimulus, 
                 group_by = Sample_Tx))+
  geom_col()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

#Total Mapped Reads per Sample
map_reads_sample_plot <- map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y = Mapped, 
                 fill = Stimulus, 
                 group_by = Sample_Tx))+
  geom_col()+
 # geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads by Sample Boxplot####
total_reads_sample_boxplot <- map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y= Total_Reads, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads by Sample Boxplot####
map_reads_sample_boxplot <- map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y= Mapped, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Reads by Stimulus####
total_reads_stimulus_plot <- map_align_stats %>% 
  ggplot(., aes (x = Stimulus, y= Total_Reads, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by stimulus"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Mapped Reads by Stimulus####
map_reads_stimulus_plot <- map_align_stats %>% 
  ggplot(., aes (x = Stimulus, y= Mapped, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by stimulus"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Reads Per Individual####
total_reads_ind_plot <- map_align_stats %>% 
  ggplot(., aes (x =as.factor(Line), y = Total_Reads))+
  geom_boxplot(aes(fill= Ind))+
 scale_fill_manual(values = ind_col)+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads Per Individual####
map_reads_ind_plot <- map_align_stats %>% 
  ggplot(., aes (x =as.factor(Line), y = Mapped))+
  geom_boxplot(aes(fill=Ind))+
 scale_fill_manual(values = c(ind_col))+
  ggtitle(expression("Total number of mapped reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Reads Per Timepoint####
map_align_stats$Time <- factor(map_align_stats$Time, 
                            levels = c("2", "24"))

reads_by_time <- c("2" = "#392344", "24" = "#457291")

total_reads_time_plot <- map_align_stats %>% 
  ggplot(., aes (x = Sample_ID, y = Total_Reads, fill = Time, group_by = Ind))+
  geom_col()+
 # geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values = reads_by_time)+
  ggtitle(expression("Total number of reads by timepoint"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Reads Per Timepoint Boxplot####
total_reads_time_boxplot <- map_align_stats %>% 
  ggplot(., aes (x = Time, y= Total_Reads, fill = Time))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_time)+
  ggtitle(expression("Total number of reads by timepoint"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads Per Timepoint####
map_reads_time_plot <- map_align_stats %>% 
  ggplot(., aes (x = Time, y= Mapped, fill = Time))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_time)+
  ggtitle(expression("Total number of mapped reads by timepoint"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Reads Per Species####
reads_by_species <- c("C" = "#E56473", "H" = "#EE4902")
total_reads_species_plot <- map_align_stats %>% 
  ggplot(., aes (x = Line, y = Total_Reads, fill = Species, group_by = Line))+
  geom_col()+
 scale_fill_manual(values=reads_by_species)+
  ggtitle(expression("Total number of reads by species"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads by Species####
total_reads_species_boxplot <- map_align_stats %>% 
  ggplot(., aes (x = Line, y= Total_Reads, fill = Species))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_species)+
  ggtitle(expression("Total number of reads by species"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads by Species####
map_reads_species_plot <- map_align_stats %>% 
  ggplot(., aes (x = Line, y= Mapped, fill = Species))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_species)+
  ggtitle(expression("Total number of mapped reads by species"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

#make a list so I can save all of these plots
# qc_map_plots_list <- list(
#   total_reads_sample_plot,
#   map_reads_sample_plot,
#   total_reads_sample_boxplot,
#   map_reads_sample_boxplot,
#   total_reads_ind_plot,
#   map_reads_ind_plot,
#   total_reads_species_plot,
#   total_reads_species_boxplot,
#   map_reads_species_plot,
#   total_reads_stimulus_plot,
#   map_reads_stimulus_plot,
#   total_reads_time_plot,
#   total_reads_time_boxplot,
#   map_reads_time_plot
# )
# 
# qc_map_plots_names <- list(
#   "total_reads_sample",
#   "map_reads_sample",
#   "total_reads_sample_boxplot",
#   "map_reads_sample_boxplot",
#   "total_reads_ind",
#   "map_reads_ind",
#   "total_reads_species",
#   "total_reads_species_boxplot",
#   "map_reads_species",
#   "total_reads_stimulus",
#   "map_reads_stimulus",
#   "total_reads_time_plot",
#   "total_reads_time_boxplot",
#   "map_reads_time"
# )
# 
# qc_map_plots_names <- paste0(qc_map_plots_names, "_EMP")
# 
# 
# stopifnot(length(qc_map_plots_list) == length(qc_map_plots_names))

#save all of these plots to my designated folder

# for (i in seq_along(qc_map_plots_list)) {
#   save_plot(
#     plot = qc_map_plots_list[[i]],
#     filename = qc_map_plots_names[i],
#     folder = output_folder
#   )
# }

```

###QC Boxplots featureCounts
```{r featureCounts Stats}

#Total Counts per Sample
map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y = Total_Counts, 
                 fill = Stimulus, 
                 group_by = Sample_Tx))+
  geom_col()+
 # geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of counts by sample"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = "white"))

#Total Assigned Counts per Sample
map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y = Assigned, 
                 fill = Stimulus, 
                 group_by = Sample_Tx))+
  geom_col()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of assigned counts by sample"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Counts by Sample Boxplot####
map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y= Total_Counts, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of counts by sample"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Assigned Counts by Sample Boxplot####
map_align_stats %>% 
  ggplot(., aes (x = Sample_Tx, y= Assigned, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of assigned counts by sample"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Counts by Stimulus####
map_align_stats %>% 
  ggplot(., aes (x = Stimulus, y= Total_Counts, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of counts by stimulus"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Assigned Counts by Stimulus####
map_align_stats %>% 
  ggplot(., aes (x = Stimulus, y= Assigned, fill = Stimulus))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of assigned counts by stimulus"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Counts Per Individual####
map_align_stats %>% 
  ggplot(., aes (x =as.factor(Line), y = Total_Counts))+
  geom_boxplot(aes(fill= Ind))+
 scale_fill_manual(values = ind_col)+
  ggtitle(expression("Total number of counts by individual"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Assigned Counts Per Individual####
map_align_stats %>% 
  ggplot(., aes (x =as.factor(Line), y = Assigned))+
  geom_boxplot(aes(fill=Ind))+
 scale_fill_manual(values = c(ind_col))+
  ggtitle(expression("Total number of assigned counts by individual"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Counts Per Timepoint####
map_align_stats$Time <- factor(map_align_stats$Time, 
                            levels = c("2", "24"))

reads_by_time <- c("2" = "#392344", "24" = "#457291")

map_align_stats %>% 
  ggplot(., aes (x = Sample_ID, y = Total_Counts, fill = Time, group_by = Ind))+
  geom_col()+
 scale_fill_manual(values = reads_by_time)+
  ggtitle(expression("Total number of counts by timepoint"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Assigned Counts Per Timepoint####
map_align_stats %>% 
  ggplot(., aes (x = Time, y= Assigned, fill = Time))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_time)+
  ggtitle(expression("Total number of assigned counts by timepoint"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Counts Per Species####
reads_by_species <- c("C" = "#E56473", "H" = "#EE4902")
map_align_stats %>% 
  ggplot(., aes (x = Line, y = Total_Counts, fill = Species, group_by = Line))+
  geom_col()+
 scale_fill_manual(values=reads_by_species)+
  ggtitle(expression("Total number of counts by species"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Total Assign Counts by Species####
map_align_stats %>% 
  ggplot(., aes (x = Line, y= Assigned, fill = Species))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_species)+
  ggtitle(expression("Total number of assigned counts by species"))+
  xlab("")+
  ylab(expression("fC Counts"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


```


#Filter Dataframe
```{r Filter with rowMeans}
#first, convert to log2cpm

#ensure that Entrez_ID are the rownames

hc_cpm_unfilt <- cpm(hc_counts, log = TRUE)
dim(hc_cpm_unfilt)
#28802 genes of 307 columns as expected

hist(hc_cpm_unfilt,
     main = "Unfiltered cpm",
     xlab = expression("log"[2]*"cpm"),
     col = 4)

#now filter by rowMeans > 0 to exclude lowly expressed genes
hc_cpm_matrix <- subset(hc_cpm_unfilt, (rowMeans(hc_cpm_unfilt) > 0))
dim(hc_cpm_matrix)
#13661 genes after filtering

#save this as an RDS and csv
# saveRDS(hc_cpm_matrix, "data/counts/hc_cpm_filtered_matrix.RDS")
# write.csv(hc_cpm_matrix, "data/counts/hc_cpm_filtered_matrix.csv")

hc_cpm_matrix <- readRDS("data/counts/hc_cpm_filtered_matrix.RDS")

#plot this out to show the filtering
hist(hc_cpm_matrix, 
     main = "Filtered Counts (cpm) rowMeans > 0",
     xlab = expression("log"[2]*"cpm"),
     col = 2)

```




#QC Boxplots
```{r QC Boxplots for H and C}
#make boxplots of all counts vs log2cpm filtered counts

par(mar = c(8,4,2,2))
#boxplot of unfiltered cpm matrix
hc_cpm_unfilt_boxplot <- boxplot(hc_cpm_unfilt, 
        main = "Unfiltered log2cpm", 
        names = colnames(hc_cpm_unfilt), 
        adj=1, las = 2, cex.axis = 0.7)

#set the margins so the x axis isn't cut off
par(mar = c(8,4,2,2))
#boxplot of filtered cpm matrix
hc_cpm_filt_boxplot <- boxplot(hc_cpm_matrix, 
        main = "Filtered log2cpm (rowMeans > 0)", 
        names = colnames(hc_cpm_matrix), 
        adj=1, las = 2, cex.axis = 0.7)


```



#Convert Ensembl to Entrez
```{r converting ensembl id into entrez id}
# ensembl_ids <- rownames(hc_cpm_matrix)
# ensembl_ids_unfilt <- rownames(hc_counts)

# entrez_ids_unfilt <- mapIds(
#   org.Hs.eg.db,
#   keys = ensembl_ids_unfilt,
#   column = "ENTREZID",
#   keytype = "ENSEMBL",
#   multiVals = "first"
# )

#since there was many mapping, check which ones are duplicated
# sum(duplicated(entrez_ids_unfilt))
#1701 are duplicated with entrez id

#convert to df
# hc_cpm_df <- as.data.frame(hc_cpm_matrix)
# hc_counts_df <- as.data.frame(hc_counts)

#add the Entrez_ID column to my counts dataframe
# hc_counts_df <- hc_counts_df %>%
#   dplyr::mutate(Entrez_ID = entrez_ids_unfilt)

#move it to the front so I can see it easily against Ensembl_ID
# hc_counts_df <- hc_counts_df %>%
#   dplyr::select(Entrez_ID, everything())

#now keep only unique values
# hc_counts_df <- hc_counts_df %>%
#   group_by(Entrez_ID) %>%
#   summarise(across(where(is.numeric), \(x) sum(x, na.rm = TRUE))) %>%
#   ungroup()

#remove any NA genes
# hc_counts_df <- hc_counts_df[!is.na(hc_counts_df$Entrez_ID), ]
# dim(hc_counts_df)
#this gives me a total of 13619 genes, which is within the range I desired
#when done for counts it gives me 28802 genes

#now make the rownames Entrez_ID
# hc_counts_df <- column_to_rownames(hc_counts_df, var = "Entrez_ID")

#save entrez_id version of dataframe
# saveRDS(hc_counts_df, "data/counts/hc_counts_entrez_dataframe.RDS")
# write.csv(hc_counts_df, "data/counts/hc_counts_entrez_dataframe.csv")

#read in data
hc_cpm_df <- readRDS("data/counts/hc_cpm_filt_entrez_dataframe.RDS")
hc_cpm_df_unfilt <- readRDS("data/counts/hc_cpm_unfilt_entrez_dataframe.RDS")
hc_counts_df <- readRDS("data/counts/hc_counts_entrez_dataframe.RDS")

# hc_counts <- hc_counts_df

# saveRDS(hc_counts, "data/counts/hc_counts_entrez.RDS")
# now move this above to where I was making cpm lists so I have entrez id instead
```

```{r PCA Plots}

prcomp_hc_unfilt <- prcomp(t(hc_cpm_unfilt), center =  TRUE)

prcomp_hc_filt <- prcomp(t(hc_cpm_matrix), center =  TRUE)

#read in my metadata annotations
metadata <- readRDS("data/counts/dissrt_metadata.RDS")

#add in labels for individual numbers
ind_num <- metadata$Ind_Num

drug_col <- readRDS("data/theme/stimulus_color_palette_all.RDS")

metadata$Time <- factor(metadata$Time)

metadata$Drug <- factor(
  metadata$Drug,
  levels = c("TUN", "THA", "DOX", "NUTL", "DMSO", 
             "LPS", "TNFa", "H2O", "BPA", "PFOA", "EtOH")
)

#now plot my PCA for unfiltered log2cpm
####PC1/PC2####
unfilt_PC1_PC2 <- ggplot2::autoplot(prcomp_hc_unfilt, data = metadata, colour = "Drug", shape = "Time", size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Unfiltered log"[2]*"cpm PC1/PC2")) +
  theme_custom()


####PC2/PC3####
unfilt_PC2_PC3 <- ggplot2::autoplot(prcomp_hc_unfilt, data = metadata, colour = "Drug", shape = "Time", size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Unfiltered log"[2]*"cpm PC2/PC3")) +
  theme_custom()

####PC3/PC4####
unfilt_PC3_PC4 <- ggplot2::autoplot(prcomp_hc_unfilt, data = metadata, colour = "Drug", shape = "Time", size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=ind_num, 
                           vjust = -.5, 
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Unfiltered log"[2]*"cpm PC3/PC4")) +
  theme_custom()


#Now plot my PCA for filtered log2cpm
####PC1/PC2####
filt_PC1_PC2 <- ggplot2::autoplot(prcomp_hc_filt, data = metadata, colour = "Drug", shape = "Time", size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=ind_num, 
                           vjust = -.5, 
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Filtered log"[2]*"cpm PC1/PC2")) +
  theme_custom()

####PC2/PC3####
filt_PC2_PC3 <- ggplot2::autoplot(prcomp_hc_filt, data = metadata, colour = "Drug", shape = "Time", size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=ind_num, 
                           vjust = -.5, 
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Filtered log"[2]*"cpm PC2/PC3")) +
  theme_custom()

####PC3/PC4####
filt_PC3_PC4 <- ggplot2::autoplot(prcomp_hc_filt, data = metadata, colour = "Drug", shape = "Time", size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=ind_num, 
                           vjust = -.5, 
                           max.overlaps = 100,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("Filtered log"[2]*"cpm PC2/PC3")) +
  theme_custom()

#now save all of these together in pdf and png
hc_pca_plots <- list(
  unfilt_PC1_PC2 = unfilt_PC1_PC2,
  unfilt_PC2_PC3 = unfilt_PC2_PC3,
  unfilt_PC3_PC4 = unfilt_PC3_PC4,
  filt_PC1_PC2 = filt_PC1_PC2,
  filt_PC2_PC3 = filt_PC2_PC3,
  filt_PC3_PC4 = filt_PC3_PC4
)

print(hc_pca_plots)

# for (plot_name in names(hc_pca_plots)) {
#   save_plot(
#     plot = hc_pca_plots[[plot_name]],
#     filename = paste0("PCA_HC_", plot_name, "_EMP"),
#     folder = output_folder
#   )
# }

```

##PCA 24hr
```{r PCA Plots 24hr Only}
#now filter out all of the 2hr points and plot your filtered PCA plots

#read in my metadata annotations
metadata <- readRDS("data/counts/dissrt_metadata.RDS")

metadata_24 <- metadata %>% 
  dplyr::filter(Time == "24")

#now make my matrices into only 24hr samples
hc_cpm_unfilt_24 <- hc_cpm_unfilt[, metadata_24$Sample_Tx, drop = FALSE]
hc_cpm_mat_24 <- hc_cpm_matrix[, metadata_24$Sample_Tx, drop = FALSE]
# saveRDS(hc_cpm_mat_24, "data/counts/hc_filtered_cpm_matrix_24hr.RDS")

#prcomp transformation
prcomp_hc_unfilt_24 <- prcomp(t(hc_cpm_unfilt_24), center = TRUE)
prcomp_hc_filt_24 <- prcomp(t(hc_cpm_mat_24), center =  TRUE)

#add in labels for individual numbers
ind_num <- metadata_24$Ind_Num
drug_col <- readRDS("data/theme/stimulus_color_palette_all.RDS")

metadata_24$Drug <- factor(
  metadata_24$Drug,
  levels = c("TUN", "THA", "DOX", "NUTL", "DMSO", 
             "LPS", "TNFa", "H2O", "BPA", "PFOA", "EtOH")
)

metadata_24$Species <- factor(
  metadata_24$Species, 
  levels = c("H", "C")
)

#now plot my PCA for unfiltered/filtered log2cpm
####PC1/PC2####
unfilt24_PC1_PC2 <- ggplot2::autoplot(prcomp_hc_unfilt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC1/PC2 24hr (unfilt)")) +
  theme_custom()


filt24_PC1_PC2 <- ggplot2::autoplot(prcomp_hc_filt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=1, y=2) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC1/PC2 24hr")) +
  theme_custom()

####PC2/3####
unfilt24_PC2_PC3 <- ggplot2::autoplot(prcomp_hc_unfilt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC2/PC3 24hr (unfilt)")) +
  theme_custom()


filt24_PC2_PC3 <- ggplot2::autoplot(prcomp_hc_filt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=2, y=3) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC2/PC3 24hr")) +
  theme_custom()

####PC3/4####
unfilt24_PC3_PC4 <- ggplot2::autoplot(prcomp_hc_unfilt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC3/PC4 24hr (unfilt)")) +
  theme_custom()


filt24_PC3_PC4 <- ggplot2::autoplot(prcomp_hc_filt_24, 
                                    data = metadata_24, 
                                    colour = "Drug", 
                                    shape = "Species", 
                                    size =4, x=3, y=4) +
  ggrepel::geom_text_repel(label=ind_num,
                           vjust = -.5,
                           max.overlaps = 50,
                           segment.color = "grey",
                           segment.size = 0.1) +
  scale_color_manual(values=drug_col) +
  ggtitle(expression("log"[2]*"cpm PC3/PC4 24hr")) +
  theme_custom()

#make a list of your 24hr PCA plots and print them all

pca_24_list <- list(
  unfilt24_PC1_PC2 = unfilt24_PC1_PC2,
  unfilt24_PC2_PC3 = unfilt24_PC2_PC3,
  unfilt24_PC3_PC4 = unfilt24_PC3_PC4,
  filt24_PC1_PC2 = filt24_PC1_PC2,
  filt24_PC2_PC3 = filt24_PC2_PC3,
  filt24_PC3_PC4 = filt24_PC3_PC4
)

print(pca_24_list)

# for (plot_name in names(pca_24_list)) {
#   save_plot(
#     plot = pca_24_list[[plot_name]],
#     filename = paste0("PCA_24hr_HC_", plot_name, "_EMP"),
#     folder = output_folder
#   )
# }


```


#Subset PCA
```{r PCA of Subsets}
#UPR Subset
upr_tx <- c("TUN", "THA", "DMSO")
upr_meta <- metadata %>% 
  dplyr::filter(Drug %in% upr_tx)
upr_cpm <- hc_cpm_df[, colnames(hc_cpm_df) %in% upr_meta$Sample_Tx] 
upr_cols <- c(
  "TUN" = "#E41A1C",
  "THA" = "#377EB8", 
  "DMSO" = "#999999"
)

#DDR Subset
ddr_tx <- c("DOX", "NUTL", "DMSO")
ddr_meta <- metadata %>% 
  dplyr::filter(Drug %in% ddr_tx)
ddr_cpm <- hc_cpm_df[, colnames(hc_cpm_df) %in% ddr_meta$Sample_Tx]  
ddr_cols <- c(
  "DOX" = "#4DAF4A", 
  "NUTL" = "#984EA3",
  "DMSO" = "#999999")

#IMR Subset
imr_tx <- c("LPS", "TNFa", "H2O")
imr_meta <- metadata %>% 
  dplyr::filter(Drug %in% imr_tx)
imr_cpm <- hc_cpm_df[, colnames(hc_cpm_df) %in% imr_meta$Sample_Tx]  
imr_cols <-  c(
  "LPS" = "#FFFF33", 
  "TNFa" = "#FC8D62", 
  "H2O" = "#777777")

#MMR Subset
mmr_tx <- c("BPA", "PFOA", "EtOH")
mmr_meta <- metadata %>% 
  dplyr::filter(Drug %in% mmr_tx)
mmr_cpm <- hc_cpm_df[, colnames(hc_cpm_df) %in% mmr_meta$Sample_Tx]  
mmr_cols <- c(
  "BPA" = "#F781BF", 
  "PFOA" = "#66C2A5", 
  "EtOH" = "#AAAAAA")

#now make a function to plot these PCA plots in subsets
make_pca_plots <- function(subset_name, subset_drugs, hc_matrix, metadata, ind_num, drug_col) {
  # filter metadata
  keep_idx <- metadata$Drug %in% subset_drugs
  metadata_sub <- metadata[keep_idx, ]
  ind_sub <- ind_num[keep_idx]
  
  # subset expression matrix to same samples
  hc_matrix_sub <- hc_matrix[, keep_idx]
  
  # run PCA on subset
  prcomp_sub <- prcomp(t(hc_matrix_sub), scale. = TRUE)
  
  ## PC1/PC2
  p1 <- autoplot(prcomp_sub, data = metadata_sub,
                 colour = "Drug", shape = "Time", size = 4,
                 x = 1, y = 2) +
    geom_text_repel(label = ind_sub,
                    vjust = -.5,
                    max.overlaps = 100,
                    segment.color = "grey",
                    segment.size = 0.1) +
    scale_color_manual(values = drug_col) +
    ggtitle(bquote(.(subset_name) ~ "subset: Filtered log"[2] * "cpm PC1/PC2")) +
    theme_custom()
  
  ## PC2/PC3
  p2 <- autoplot(prcomp_sub, data = metadata_sub,
                 colour = "Drug", shape = "Time", size = 4,
                 x = 2, y = 3) +
    geom_text_repel(label = ind_sub,
                    vjust = -.5,
                    max.overlaps = 100,
                    segment.color = "grey",
                    segment.size = 0.1) +
    scale_color_manual(values = drug_col) +
    ggtitle(bquote(.(subset_name) ~ "subset: Filtered log"[2] * "cpm PC2/PC3")) +
    theme_custom()
  
  list(p1 = p1, p2 = p2)
}

# ---- Run for each subset ----
upr_plots <- make_pca_plots("UPR", c("TUN", "THA", "DMSO"), hc_cpm_matrix, metadata, ind_num, drug_col)
ddr_plots <- make_pca_plots("DDR", c("DOX", "NUTL", "DMSO"), hc_cpm_matrix, metadata, ind_num, drug_col)
imr_plots <- make_pca_plots("IMR", c("LPS", "TNFa", "H2O"), hc_cpm_matrix, metadata, ind_num, drug_col)
mmr_plots <- make_pca_plots("MMR", c("BPA", "PFOA", "EtOH"), hc_cpm_matrix, metadata, ind_num, drug_col)

all_plots_subset_pca <- list(
  upr_PC1_PC2 = upr_plots$p1,
  upr_PC2_PC3 = upr_plots$p2,
  ddr_PC1_PC2 = ddr_plots$p1,
  ddr_PC2_PC3 = ddr_plots$p2,
  imr_PC1_PC2 = imr_plots$p1,
  imr_PC2_PC3 = imr_plots$p2,
  mmr_PC1_PC2 = mmr_plots$p1,
  mmr_PC2_PC3 = mmr_plots$p2
)

print(all_plots_subset_pca)

# # Loop through each plot and save
# for (plot_name in names(all_plots_subset_pca)) {
#   save_plot(
#     plot = all_plots_subset_pca[[plot_name]],
#     filename = paste0("Subset_PCA_", plot_name, "_EMP"),
#     folder = output_folder
#     )
# }


```

##24hr Subset PCA
```{r Subset PCA plots 24hr only}
#now do the same thing you did above but filter out any 2hr samples

#make matrices into only 24hr samples
hc_cpm_mat_24 <- hc_cpm_matrix[, metadata_24$Sample_Tx, drop = FALSE]

#UPR Subset
upr_tx <- c("TUN", "THA", "DMSO")
upr_meta_24 <- metadata_24 %>% 
  dplyr::filter(Drug %in% upr_tx)
upr_cpm_24 <- hc_cpm_mat_24[, colnames(hc_cpm_mat_24) %in% upr_meta_24$Sample_Tx] 

#DDR Subset
ddr_tx <- c("DOX", "NUTL", "DMSO")
ddr_meta_24 <- metadata_24 %>% 
  dplyr::filter(Drug %in% ddr_tx)
ddr_cpm_24 <- hc_cpm_mat_24[, colnames(hc_cpm_mat_24) %in% ddr_meta_24$Sample_Tx]  

#IMR Subset
imr_tx <- c("LPS", "TNFa", "H2O")
imr_meta_24 <- metadata_24 %>% 
  dplyr::filter(Drug %in% imr_tx)
imr_cpm_24 <- hc_cpm_mat_24[, colnames(hc_cpm_mat_24) %in% imr_meta_24$Sample_Tx]  

#MMR Subset
mmr_tx <- c("BPA", "PFOA", "EtOH")
mmr_meta_24 <- metadata_24 %>% 
  dplyr::filter(Drug %in% mmr_tx)
mmr_cpm_24 <- hc_cpm_mat_24[, colnames(hc_cpm_mat_24) %in% mmr_meta_24$Sample_Tx]  


#now make a function to plot these PCA plots in subsets
#use metadata not metadata_24 since it needs to be self-contained
make_pca_plots_24 <- function(subset_name, subset_drugs, hc_matrix, metadata, ind_num, drug_col) {
  # filter metadata
  keep_idx <- metadata$Drug %in% subset_drugs
  metadata_sub <- metadata[keep_idx, ]
  ind_sub <- ind_num[keep_idx]
  
  # subset expression matrix to same samples
  hc_matrix_sub <- hc_matrix[, keep_idx, drop = FALSE]
  
  drug_col_sub <- drug_col[names(drug_col) %in% unique(metadata_sub$Drug)]
  
  # run PCA on subset
  prcomp_sub <- prcomp(t(hc_matrix_sub), scale. = TRUE)
  
  ## PC1/PC2
  p1 <- autoplot(prcomp_sub, data = metadata_sub,
                 colour = "Drug", shape = "Species", size = 4,
                 x = 1, y = 2) +
    geom_text_repel(label = ind_sub,
                    vjust = -.5,
                    max.overlaps = 50,
                    segment.color = "grey",
                    segment.size = 0.1) +
    scale_color_manual(values = drug_col_sub) +
    ggtitle(bquote(.(subset_name) ~ "subset: log"[2]*"cpm PC1/PC2")) +
    theme_custom()
  
  ## PC2/PC3
  p2 <- autoplot(prcomp_sub, data = metadata_sub,
                 colour = "Drug", shape = "Species", size = 4,
                 x = 2, y = 3) +
    geom_text_repel(label = ind_sub,
                    vjust = -.5,
                    max.overlaps = 100,
                    segment.color = "grey",
                    segment.size = 0.1) +
    scale_color_manual(values = drug_col_sub) +
    ggtitle(bquote(.(subset_name) ~ "subset: log"[2]*"cpm PC2/PC3")) +
    theme_custom()
  
  list(p1 = p1, p2 = p2)
}

# ---- Run for each subset ----
upr_plots_24 <- make_pca_plots_24("UPR", c("TUN", "THA", "DMSO"), hc_cpm_mat_24, metadata_24, ind_num, drug_col)
ddr_plots_24 <- make_pca_plots_24("DDR", c("DOX", "NUTL", "DMSO"), hc_cpm_mat_24, metadata_24, ind_num, drug_col)
imr_plots_24 <- make_pca_plots_24("IMR", c("LPS", "TNFa", "H2O"), hc_cpm_mat_24, metadata_24, ind_num, drug_col)
mmr_plots_24 <- make_pca_plots_24("MMR", c("BPA", "PFOA", "EtOH"), hc_cpm_mat_24, metadata_24, ind_num, drug_col)

all_plots_subset_pca24 <- list(
  upr24_PC1_PC2 = upr_plots_24$p1,
  upr24_PC2_PC3 = upr_plots_24$p2,
  ddr24_PC1_PC2 = ddr_plots_24$p1,
  ddr24_PC2_PC3 = ddr_plots_24$p2,
  imr24_PC1_PC2 = imr_plots_24$p1,
  imr24_PC2_PC3 = imr_plots_24$p2,
  mmr24_PC1_PC2 = mmr_plots_24$p1,
  mmr24_PC2_PC3 = mmr_plots_24$p2
)

print(all_plots_subset_pca24)

# Loop through each plot and save
for (plot_name in names(all_plots_subset_pca24)) {
  save_plot(
    plot = all_plots_subset_pca24[[plot_name]],
    filename = paste0("Subset_PCA_24_", plot_name, "_EMP"),
    folder = output_folder
    )
}



```


#Correlation Heatmaps
```{r Spearman Heatmaps Correlation, fig.width=18, fig.height=14, warning=FALSE}
#first make correlation heatmaps of all samples
#using ComplexHeatmap

#First I want to alter the column names to be clearer
new_names <- metadata$Final_sample_name[match(colnames(x), metadata$Final_sample_name)]

colnames(hc_cpm_df) <- new_names

#prepare data
hc_matrix <- hc_cpm_df
colnames(hc_matrix) <- metadata$Sample_Tx  # Ensure column names match metadata

#compute Spearman correlation
cor_matrix_spearman <- cor(hc_matrix, method = "spearman", use = "everything")

#define a standard color scale where 0 = white
col_fun <- colorRamp2(
  breaks = c(-1,0,1),
  colors = c("blue", "white", "red")
)

#extract metadata for annotations
Individual <- as.character(metadata$Ind)
Species <- as.character(metadata$Species)
Time <- as.character(metadata$Time)
Stimulus <- as.character(metadata$Drug)

#define colors for my annotations
stim_col <- readRDS("data/theme/stimulus_color_palette_all.RDS")
ind_col <- readRDS("data/theme/individual_category_color_palette.RDS")
time_col <- readRDS("data/theme/time_category_color_palette.RDS")
spec_col <- readRDS("data/theme/species_category_color_palette.RDS")

#create my annotations
top_annotation <- HeatmapAnnotation(
  Individual = Individual, 
  Time = Time,
  Stimulus = Stimulus,
  Species = Species,
  col = list(
    Individual = ind_col,
    Time = time_col,
    Stimulus = stim_col,
    Species = spec_col
  )
)

# ---- Spearman heatmap for all samples ----
heatmap_spearman_all <- Heatmap(cor_matrix_spearman,
                                name = "Spearman",
                                col = col_fun,
                                top_annotation = top_annotation,
                                show_row_names = TRUE,
                                show_column_names = TRUE,
                                cluster_rows = TRUE,
                                cluster_columns = TRUE,
                                border = TRUE, 
                                show_row_dend = FALSE,
                                show_column_dend = FALSE,
                                use_raster = TRUE,
                                raster_quality = 2,
                                raster_device = "CairoPNG",
                                row_names_gp = gpar(fontsize = 8),
                                column_names_gp = gpar(fontsize = 8),
                                cell_fun = function(j, i, x, y, 
                                                    width, 
                                                    height, 
                                                    fill) {
          #draw a black border
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
        })

# draw(heatmap_spearman_all)

# ---- Define subsets ----
subsets <- list(
  UPR = c("TUN", "THA", "DMSO"),
  DDR = c("DOX", "NUTL", "DMSO"),
  IMR = c("LPS", "TNFa", "H2O"),
  MMR = c("BPA", "PFOA", "EtOH")
)

# ---- Loop through subsets ----
subset_heatmaps <- list()

for (subset_name in names(subsets)) {
  # Filter columns for this subset
  cols_subset <- metadata$Sample_Tx[metadata$Drug %in% subsets[[subset_name]]]
  mat_subset <- hc_matrix[, cols_subset, drop = FALSE]
  
  # Correlation
  cor_subset <- cor(mat_subset, method = "spearman", use = "everything")
  
  # Annotations for this subset
  top_ann_subset <- HeatmapAnnotation(
    Individual = metadata$Ind[metadata$Sample_Tx %in% cols_subset],
    Time = metadata$Time[metadata$Sample_Tx %in% cols_subset],
    Stimulus = metadata$Drug[metadata$Sample_Tx %in% cols_subset],
    Species = metadata$Species[metadata$Sample_Tx %in% cols_subset],
    col = list(
      Individual = ind_col,
      Time = time_col,
      Stimulus = stim_col,
      Species = spec_col
    )
  )
  
  # Heatmap
  heatmap_subset <- Heatmap(cor_subset,
                            name = paste0("Spearman_", subset_name),
                            top_annotation = top_ann_subset,
                            col = col_fun,
                            show_row_names = TRUE,
                            show_column_names = TRUE,
                            cluster_rows = TRUE,
                            cluster_columns = TRUE,
                            border = TRUE,
                            show_row_dend = FALSE,
                            show_column_dend = FALSE,
                            use_raster = TRUE,
                            raster_quality = 2,
                            raster_device = "CairoPNG",
                            row_names_gp = gpar(fontsize = 8),
                            column_names_gp = gpar(fontsize = 8), 
                            cell_fun = function(j, i, 
                                                x, y, 
                                                width, height, fill) {
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
        })
  
  
  # draw(heatmap_subset)
  subset_heatmaps[[subset_name]] <- heatmap_subset
}

#save my subset heatmaps
for (nm in names(subset_heatmaps)) {
  ht <- subset_heatmaps[[nm]]
  save_complex_heatmap(
    ht = ht,
    filename = paste0("CorHM_Spearman_", nm, "_EMP"),
    folder = output_folder,
    height = 18,
    width = 18
  )
  message("Saved subset heatmap: ", nm)
}

#save all samples heatmap
save_complex_heatmap(
  ht = heatmap_spearman_all,
  filename = "CorHM_Spearman_AllSamples_EMP",
  folder = output_folder,
  height = 40, width = 40)


```

##24hr Correlation Heatmaps
```{r Cor HMs Spearman 24hr Only}
#first make correlation heatmaps of all samples
#using ComplexHeatmap

hc_cpm_df_24 <- hc_cpm_df[, metadata_24$Final_sample_name, drop = FALSE]

#First I want to alter the column names so they're not inc time
clean_sample_names <- function(x) {
  #changing col names DOX_24_H_1 -> DOX_H1
  x <- gsub("_24_", "_", x)      
  x <- gsub("_([HC])_", "_\\1", x) 
  x <- gsub("_(\\d+)$", "\\1", x)
  return(x)
}

# Apply only to 24hr samples in metadata_24
meta_24 <- metadata_24 %>%
  mutate(Sample_Tx_clean = ifelse(
    grepl("_24_", Sample_Tx),
    clean_sample_names(Sample_Tx),
    Sample_Tx
  ))

colnames(hc_cpm_df_24) <- clean_sample_names(colnames(hc_cpm_df_24))

#prepare data
hc_matrix_24 <- hc_cpm_df_24
colnames(hc_matrix_24) <- meta_24$Sample_Tx_clean

#compute Spearman correlation
cor_matrix_spearman_24 <- cor(hc_matrix_24, method = "spearman", use = "everything")

#define a standard color scale where 0 = white
col_fun <- colorRamp2(
  breaks = c(-1,0,1),
  colors = c("blue", "white", "red")
)

#extract metadata for annotations
Individual <- as.character(meta_24$Ind)
Species <- as.character(meta_24$Species)
Stimulus <- as.character(meta_24$Drug)
Response <- as.character(meta_24$Response)


#define colors for my annotations
stim_col <- readRDS("data/theme/stimulus_color_palette_all.RDS")
ind_col <- readRDS("data/theme/individual_category_color_palette.RDS")
spec_col <- readRDS("data/theme/species_category_color_palette.RDS")
resp_col <- readRDS("data/theme/response_category_color_palette.RDS")

#create my annotations
top_annotation_24 <- HeatmapAnnotation(
  Individual = Individual, 
  Stimulus = Stimulus,
  Species = Species,
  Response = Response,
  col = list(
    Individual = ind_col,
    Stimulus = stim_col,
    Species = spec_col,
    Response = resp_col
  )
)

#spearman heatmap for all samples
heatmap_spearman_24 <- Heatmap(cor_matrix_spearman_24,
                                name = "Spearman",
                                col = col_fun,
                                top_annotation = top_annotation_24,
                                show_row_names = TRUE,
                                show_column_names = TRUE,
                                cluster_rows = TRUE,
                                cluster_columns = TRUE,
                                border = TRUE, 
                                show_row_dend = FALSE,
                                show_column_dend = FALSE,
                                use_raster = TRUE,
                                raster_quality = 2,
                                raster_device = "CairoPNG",
                                row_names_gp = gpar(fontsize = 8),
                                column_names_gp = gpar(fontsize = 8),
                                layer_fun = function(j, i, x, y, 
                                                    width, 
                                                    height, 
                                                    fill) {
          #draw a black border
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
        })

# draw(heatmap_spearman_24)

# ---- Define subsets ----
subsets <- list(
  UPR = c("TUN", "THA", "DMSO"),
  DDR = c("DOX", "NUTL", "DMSO"),
  IMR = c("LPS", "TNFa", "H2O"),
  MMR = c("BPA", "PFOA", "EtOH")
)

# ---- Loop through subsets ----
subset_heatmaps_24 <- list()

for (subset_name in names(subsets)) {
  # Filter columns for this subset
  cols_subset_24 <- meta_24$Sample_Tx_clean[meta_24$Drug %in% subsets[[subset_name]]]
  mat_subset <- hc_matrix_24[, cols_subset_24, drop = FALSE]
  
  # Correlation
  cor_subset_24 <- cor(mat_subset, method = "spearman", use = "everything")
  
  # Annotations for this subset
  top_ann_subset_24 <- HeatmapAnnotation(
    Individual = meta_24$Ind[meta_24$Sample_Tx_clean %in% cols_subset_24],
    Stimulus = meta_24$Drug[meta_24$Sample_Tx_clean %in% cols_subset_24],
    Species = meta_24$Species[meta_24$Sample_Tx_clean %in% cols_subset_24],
    col = list(
      Individual = ind_col,
      Stimulus = stim_col,
      Species = spec_col
    )
  )
  
  # Heatmap
  heatmap_subset_24 <- Heatmap(cor_subset_24,
                            name = paste0("Spearman_24_", subset_name),
                            top_annotation = top_ann_subset_24,
                            col = col_fun,
                            show_row_names = TRUE,
                            show_column_names = TRUE,
                            cluster_rows = TRUE,
                            cluster_columns = TRUE,
                            border = TRUE,
                            show_row_dend = FALSE,
                            show_column_dend = FALSE,
                            use_raster = TRUE,
                            raster_quality = 2,
                            raster_device = "CairoPNG",
                            row_names_gp = gpar(fontsize = 8),
                            column_names_gp = gpar(fontsize = 8), 
                            layer_fun = function(j, i, 
                                                x, y, 
                                                width, height, fill) {
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
        })
  
  
  # draw(heatmap_subset_24)
  subset_heatmaps_24[[subset_name]] <- heatmap_subset_24
}

#save my subset heatmaps
for (nm in names(subset_heatmaps_24)) {
  ht <- subset_heatmaps_24[[nm]]
  save_complex_heatmap(
    ht = ht,
    filename = paste0("CorHM_Spearman_24_", nm, "_EMP_251002"),
    folder = output_folder,
    height = 12,
    width = 12
  )
  message("Saved subset heatmap: ", nm)
}

#save all samples heatmap
save_complex_heatmap(
  ht = heatmap_spearman_24,
  filename = "CorHM_Spearman_24hr_EMP_251002",
  folder = output_folder,
  height = 18,
  width = 18
)

```


#Perform Differential Expression Analysis
```{r DE Analysis}
#filter my counts for DE
#make sure that hc_counts with my raw counts information has the right column names
# identical(colnames(hc_counts_df), colnames(hc_cpm_df))
#TRUE

# x = hc_counts_df[row.names(hc_cpm_df), ] %>%
#   as.matrix()
#now i've filtered my raw counts for DE by the number of genes that I got after rowMeans filtering

#modify my metadata to match with the rownames being my sample names in columns here
# metadata_2 <- metadata %>%
#   as.data.frame()

rownames(metadata_2) <- metadata_2$Sample_ID
colnames(x) <- metadata_2$Final_sample_name
rownames(metadata_2) <- metadata_2$Final_sample_name

metadata_2$Cond <- make.names(metadata_2$Cond)
metadata_2$Ind <- as.character(metadata_2$Ind)

# saveRDS(metadata_2, "data/de/metadata_2_de.RDS")
# saveRDS(x, "data/de/x_counts_de.RDS")

#read in your saved metadata_2 and x so it doesn't get altered
metadata_2 <- readRDS("data/de/metadata_2_de.RDS")
x <- readRDS("data/de/x_counts_de.RDS")

#now that I've formatted my dataframes, I'm going to go ahead and start the process of DE

#create DGEList object
dge <- DGEList(counts = x)
#make sure that the groups are based on my column names
dge$samples$group <- factor(metadata_2$Cond)
#now perform TMM normalization for library size
dge <- calcNormFactors(dge, method = "TMM")

#before making my design matrix, factor
metadata_2$Ind_Num <- factor(metadata_2$Ind_Num, levels = unique(metadata_2$Ind_Num))
metadata_2$Drug <- factor(metadata_2$Drug, levels = c("TUN", "THA", "DOX", "NUTL", "DMSO", "LPS", "TNFa", "H2O", "BPA", "PFOA", "EtOH"))
metadata_2$Time <- factor(metadata_2$Time, levels = c("2", "24"))
metadata_2$Species <- factor(metadata_2$Species, levels = c("C", "H"))
metadata_2$Cond <- factor(metadata_2$Cond, levels = unique(metadata_2$Cond))

#create my design matrix
design <- model.matrix(~ 0 + metadata_2$Cond)
colnames(design) <- gsub("metadata_2\\$Cond", "", colnames(design))

# saveRDS(design, "data/de/design_matrix.RDS")

#since I factored condition above, they stayed in the right order but keep an eye on them

#run duplicate correlation for individual effect
corfit <- duplicateCorrelation(object = dge$counts,
                               design = design,
                               block = metadata_2$Ind_Num)

#run voom transformation and plot it
v <- voom(dge, design, 
          block = metadata_2$Ind_Num, 
          correlation = corfit$consensus.correlation)

#fit my linear model
fit <- lmFit(v, design,
             block = metadata_2$Ind_Num,
             correlation = corfit$consensus.correlation)

#put together my contrast matrix
contrast_matrix <- makeContrasts(
  V.TUN2_C = TUN_2_C - DMSO_2_C,
  V.THA2_C = THA_2_C - DMSO_2_C,
  V.DOX2_C = DOX_2_C - DMSO_2_C,
  V.NUTL2_C = NUTL_2_C - DMSO_2_C,
  V.LPS2_C = LPS_2_C - H2O_2_C,
  V.TNFa2_C = TNFa_2_C - H2O_2_C,
  V.BPA2_C = BPA_2_C - EtOH_2_C,
  V.PFOA2_C = PFOA_2_C - EtOH_2_C,
  V.TUN24_C = TUN_24_C - DMSO_24_C,
  V.THA24_C = THA_24_C - DMSO_24_C,
  V.DOX24_C = DOX_24_C - DMSO_24_C,
  V.NUTL24_C = NUTL_24_C - DMSO_24_C,
  V.LPS24_C = LPS_24_C - H2O_24_C,
  V.TNFa24_C = TNFa_24_C - H2O_24_C,
  V.BPA24_C = BPA_24_C - EtOH_24_C,
  V.PFOA24_C = PFOA_24_C - EtOH_24_C,
  V.TUN2_H = TUN_2_H - DMSO_2_H,
  V.THA2_H = THA_2_H - DMSO_2_H,
  V.DOX2_H = DOX_2_H - DMSO_2_H,
  V.NUTL2_H = NUTL_2_H - DMSO_2_H,
  V.LPS2_H = LPS_2_H - H2O_2_H,
  V.TNFa2_H = TNFa_2_H - H2O_2_H,
  V.BPA2_H = BPA_2_H - EtOH_2_H,
  V.PFOA2_H = PFOA_2_H - EtOH_2_H,
  V.TUN24_H = TUN_24_H - DMSO_24_H,
  V.THA24_H = THA_24_H - DMSO_24_H,
  V.DOX24_H = DOX_24_H - DMSO_24_H,
  V.NUTL24_H = NUTL_24_H - DMSO_24_H,
  V.LPS24_H = LPS_24_H - H2O_24_H,
  V.TNFa24_H = TNFa_24_H - H2O_24_H,
  V.BPA24_H = BPA_24_H - EtOH_24_H,
  V.PFOA24_H = PFOA_24_H - EtOH_24_H,
  levels = design
)

#apply these contrasts to compare drug to matched vehicles
fit2 <- contrasts.fit(fit, contrast_matrix)
fit2 <- eBayes(fit2)

#plot the mean-variance trend
plotSA(fit2, main = "Dissertation Model: MV Trend")

#pull my results summary for DEGs
results_summary <- decideTests(fit2, 
                               adjust.method = "BH",
                               p.value = 0.05)

summary(results_summary)

```

#Create Toptables
```{r DE Toptables Chimp}
#make toptables of DEGs - start with Chimp

#2hr Chimp
Toptable_V.TUN2_C <- topTable(fit = fit2, 
                            coef = "V.TUN2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.THA2_C <- topTable(fit = fit2, 
                            coef = "V.THA2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.DOX2_C <- topTable(fit = fit2, 
                            coef = "V.DOX2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.NUTL2_C <- topTable(fit = fit2, 
                            coef = "V.NUTL2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.LPS2_C <- topTable(fit = fit2, 
                            coef = "V.LPS2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.TNFa2_C <- topTable(fit = fit2, 
                            coef = "V.TNFa2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.BPA2_C <- topTable(fit = fit2, 
                            coef = "V.BPA2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.PFOA2_C <- topTable(fit = fit2, 
                            coef = "V.PFOA2_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

#24hr toptables
Toptable_V.TUN24_C <- topTable(fit = fit2, 
                            coef = "V.TUN24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.THA24_C <- topTable(fit = fit2, 
                            coef = "V.THA24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.DOX24_C <- topTable(fit = fit2, 
                            coef = "V.DOX24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.NUTL24_C <- topTable(fit = fit2, 
                            coef = "V.NUTL24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.LPS24_C <- topTable(fit = fit2, 
                            coef = "V.LPS24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.TNFa24_C <- topTable(fit = fit2, 
                            coef = "V.TNFa24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.BPA24_C <- topTable(fit = fit2, 
                            coef = "V.BPA24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.PFOA24_C <- topTable(fit = fit2, 
                            coef = "V.PFOA24_C",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

```

```{r DE Toptables Human}
#2hr Human
Toptable_V.TUN2_H <- topTable(fit = fit2, 
                            coef = "V.TUN2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.THA2_H <- topTable(fit = fit2, 
                            coef = "V.THA2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.DOX2_H <- topTable(fit = fit2, 
                            coef = "V.DOX2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.NUTL2_H <- topTable(fit = fit2, 
                            coef = "V.NUTL2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.LPS2_H <- topTable(fit = fit2, 
                            coef = "V.LPS2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.TNFa2_H <- topTable(fit = fit2, 
                            coef = "V.TNFa2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.BPA2_H <- topTable(fit = fit2, 
                            coef = "V.BPA2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.PFOA2_H <- topTable(fit = fit2, 
                            coef = "V.PFOA2_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

#24hr toptables
Toptable_V.TUN24_H <- topTable(fit = fit2, 
                            coef = "V.TUN24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.THA24_H <- topTable(fit = fit2, 
                            coef = "V.THA24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.DOX24_H <- topTable(fit = fit2, 
                            coef = "V.DOX24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.NUTL24_H <- topTable(fit = fit2, 
                            coef = "V.NUTL24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.LPS24_H <- topTable(fit = fit2, 
                            coef = "V.LPS24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.TNFa24_H <- topTable(fit = fit2, 
                            coef = "V.TNFa24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.BPA24_H <- topTable(fit = fit2, 
                            coef = "V.BPA24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

Toptable_V.PFOA24_H <- topTable(fit = fit2, 
                            coef = "V.PFOA24_H",
                            number = nrow(x), 
                            adjust.method = "BH", 
                            p.value = 1, 
                            sort.by = "none")

```

#Combine Toptables List
```{r Combined Toptables}

c_toptables_2 <- list(
  "V.TUN2_C" = Toptable_V.TUN2_C, 
  "V.THA2_C" = Toptable_V.THA2_C, 
  "V.DOX2_C" = Toptable_V.DOX2_C,
  "V.NUTL2_C" = Toptable_V.NUTL2_C, 
  "V.LPS2_C" = Toptable_V.LPS2_C, 
  "V.TNFa2_C" = Toptable_V.TNFa2_C,
  "V.BPA2_C" = Toptable_V.BPA2_C,
  "V.PFOA2_C" = Toptable_V.PFOA2_C
)

h_toptables_2 <- list(
  "V.TUN2_H" = Toptable_V.TUN2_H, 
  "V.THA2_H" = Toptable_V.THA2_H, 
  "V.DOX2_H" = Toptable_V.DOX2_H,
  "V.NUTL2_H" = Toptable_V.NUTL2_H, 
  "V.LPS2_H" = Toptable_V.LPS2_H, 
  "V.TNFa2_H" = Toptable_V.TNFa2_H,
  "V.BPA2_H" = Toptable_V.BPA2_H,
  "V.PFOA2_H" = Toptable_V.PFOA2_H
)

c_toptables_24 <- list(
  "V.TUN24_C" = Toptable_V.TUN24_C,
  "V.THA24_C" = Toptable_V.THA24_C,
  "V.DOX24_C" = Toptable_V.DOX24_C,
  "V.NUTL24_C" = Toptable_V.NUTL24_C,
  "V.LPS24_C" = Toptable_V.LPS24_C,
  "V.TNFa24_C" = Toptable_V.TNFa24_C,
  "V.BPA24_C" = Toptable_V.BPA24_C,
  "V.PFOA24_C" = Toptable_V.PFOA24_C
)

h_toptables_24 <- list(
  "V.TUN24_H" = Toptable_V.TUN24_H,
  "V.THA24_H" = Toptable_V.THA24_H,
  "V.DOX24_H" = Toptable_V.DOX24_H,
  "V.NUTL24_H" = Toptable_V.NUTL24_H,
  "V.LPS24_H" = Toptable_V.LPS24_H,
  "V.TNFa24_H" = Toptable_V.TNFa24_H,
  "V.BPA24_H" = Toptable_V.BPA24_H,
  "V.PFOA24_H" = Toptable_V.PFOA24_H
)

c_toptables_all <- list(
  "V.TUN2_C" = Toptable_V.TUN2_C, 
  "V.THA2_C" = Toptable_V.THA2_C, 
  "V.DOX2_C" = Toptable_V.DOX2_C,
  "V.NUTL2_C" = Toptable_V.NUTL2_C, 
  "V.LPS2_C" = Toptable_V.LPS2_C, 
  "V.TNFa2_C" = Toptable_V.TNFa2_C,
  "V.BPA2_C" = Toptable_V.BPA2_C,
  "V.PFOA2_C" = Toptable_V.PFOA2_C,
  "V.TUN24_C" = Toptable_V.TUN24_C,
  "V.THA24_C" = Toptable_V.THA24_C,
  "V.DOX24_C" = Toptable_V.DOX24_C,
  "V.NUTL24_C" = Toptable_V.NUTL24_C,
  "V.LPS24_C" = Toptable_V.LPS24_C,
  "V.TNFa24_C" = Toptable_V.TNFa24_C,
  "V.BPA24_C" = Toptable_V.BPA24_C,
  "V.PFOA24_C" = Toptable_V.PFOA24_C
)

h_toptables_all <- list(
  "V.TUN2_H" = Toptable_V.TUN2_H, 
  "V.THA2_H" = Toptable_V.THA2_H, 
  "V.DOX2_H" = Toptable_V.DOX2_H,
  "V.NUTL2_H" = Toptable_V.NUTL2_H, 
  "V.LPS2_H" = Toptable_V.LPS2_H, 
  "V.TNFa2_H" = Toptable_V.TNFa2_H,
  "V.BPA2_H" = Toptable_V.BPA2_H,
  "V.PFOA2_H" = Toptable_V.PFOA2_H,
  "V.TUN24_H" = Toptable_V.TUN24_H,
  "V.THA24_H" = Toptable_V.THA24_H,
  "V.DOX24_H" = Toptable_V.DOX24_H,
  "V.NUTL24_H" = Toptable_V.NUTL24_H,
  "V.LPS24_H" = Toptable_V.LPS24_H,
  "V.TNFa24_H" = Toptable_V.TNFa24_H,
  "V.BPA24_H" = Toptable_V.BPA24_H,
  "V.PFOA24_H" = Toptable_V.PFOA24_H
)

hc_toptables <- list(
  "V.TUN2_H" = Toptable_V.TUN2_H,
  "V.THA2_H" = Toptable_V.THA2_H,
  "V.DOX2_H" = Toptable_V.DOX2_H,
  "V.NUTL2_H" = Toptable_V.NUTL2_H,
  "V.LPS2_H" = Toptable_V.LPS2_H,
  "V.TNFa2_H" = Toptable_V.TNFa2_H,
  "V.BPA2_H" = Toptable_V.BPA2_H,
  "V.PFOA2_H" = Toptable_V.PFOA2_H,
  "V.TUN24_H" = Toptable_V.TUN24_H,
  "V.THA24_H" = Toptable_V.THA24_H,
  "V.DOX24_H" = Toptable_V.DOX24_H,
  "V.NUTL24_H" = Toptable_V.NUTL24_H,
  "V.LPS24_H" = Toptable_V.LPS24_H,
  "V.TNFa24_H" = Toptable_V.TNFa24_H,
  "V.BPA24_H" = Toptable_V.BPA24_H,
  "V.PFOA24_H" = Toptable_V.PFOA24_H,
  "V.TUN2_C" = Toptable_V.TUN2_C,
  "V.THA2_C" = Toptable_V.THA2_C,
  "V.DOX2_C" = Toptable_V.DOX2_C,
  "V.NUTL2_C" = Toptable_V.NUTL2_C,
  "V.LPS2_C" = Toptable_V.LPS2_C,
  "V.TNFa2_C" = Toptable_V.TNFa2_C,
  "V.BPA2_C" = Toptable_V.BPA2_C,
  "V.PFOA2_C" = Toptable_V.PFOA2_C,
  "V.TUN24_C" = Toptable_V.TUN24_C,
  "V.THA24_C" = Toptable_V.THA24_C,
  "V.DOX24_C" = Toptable_V.DOX24_C,
  "V.NUTL24_C" = Toptable_V.NUTL24_C,
  "V.LPS24_C" = Toptable_V.LPS24_C,
  "V.TNFa24_C" = Toptable_V.TNFa24_C,
  "V.BPA24_C" = Toptable_V.BPA24_C,
  "V.PFOA24_C" = Toptable_V.PFOA24_C
)


#save each of these toptables in a list so I can access them later for analysis 
# saveRDS(list(
#   "V.TUN2_H" = Toptable_V.TUN2_H,
#   "V.THA2_H" = Toptable_V.THA2_H,
#   "V.DOX2_H" = Toptable_V.DOX2_H,
#   "V.NUTL2_H" = Toptable_V.NUTL2_H,
#   "V.LPS2_H" = Toptable_V.LPS2_H,
#   "V.TNFa2_H" = Toptable_V.TNFa2_H,
#   "V.BPA2_H" = Toptable_V.BPA2_H,
#   "V.PFOA2_H" = Toptable_V.PFOA2_H,
#   "V.TUN24_H" = Toptable_V.TUN24_H,
#   "V.THA24_H" = Toptable_V.THA24_H,
#   "V.DOX24_H" = Toptable_V.DOX24_H,
#   "V.NUTL24_H" = Toptable_V.NUTL24_H,
#   "V.LPS24_H" = Toptable_V.LPS24_H,
#   "V.TNFa24_H" = Toptable_V.TNFa24_H,
#   "V.BPA24_H" = Toptable_V.BPA24_H,
#   "V.PFOA24_H" = Toptable_V.PFOA24_H,
#   "V.TUN2_C" = Toptable_V.TUN2_C,
#   "V.THA2_C" = Toptable_V.THA2_C,
#   "V.DOX2_C" = Toptable_V.DOX2_C,
#   "V.NUTL2_C" = Toptable_V.NUTL2_C,
#   "V.LPS2_C" = Toptable_V.LPS2_C,
#   "V.TNFa2_C" = Toptable_V.TNFa2_C,
#   "V.BPA2_C" = Toptable_V.BPA2_C,
#   "V.PFOA2_C" = Toptable_V.PFOA2_C,
#   "V.TUN24_C" = Toptable_V.TUN24_C,
#   "V.THA24_C" = Toptable_V.THA24_C,
#   "V.DOX24_C" = Toptable_V.DOX24_C,
#   "V.NUTL24_C" = Toptable_V.NUTL24_C,
#   "V.LPS24_C" = Toptable_V.LPS24_C,
#   "V.TNFa24_C" = Toptable_V.TNFa24_C,
#   "V.BPA24_C" = Toptable_V.BPA24_C,
#   "V.PFOA24_C" = Toptable_V.PFOA24_C
# ), file = "data/de/Toptable_list.RDS")


#save each of these toptables for each comparison
#convert rownames to Entrez_ID for every topTable
# toptable_list_entrez <- lapply(toptable_list, function(df) {
#   df$Entrez_ID <- rownames(df)
#   df <- df[, c("Entrez_ID", setdiff(colnames(df), "Entrez_ID"))]  # Entrez_ID first
#   return(df)
# })
# 
# #save csv 
# for (nm in names(toptable_list_entrez)) {
#   write.csv(
#     toptable_list_entrez[[nm]],
#     file = file.path("data/de", paste0("Toptable_", nm, ".csv")),
#     row.names = FALSE
#   )
# }

#save as an RDS as well
# saveRDS(toptable_list_entrez, file = "data/de/Toptable_list.RDS")

#now go ahead and filter for adj. p value < 0.05 to make a list of all significant DEGs in these lists per comparison
# toptable_sig_list <- list()

# #make a function to loop through toptables
# for (nm in names(toptable_list)) {
#   df <- toptable_list[[nm]]
#   
#   #find the adj. p value to filter by
#   if ("adj.P.Val" %in% colnames(df)) {
#     df_sig <- df[df$adj.P.Val < 0.05, ]
#   } else {
#     warning(paste("No adj. p-value column in", nm))
#     next
#   }
#   
#   #skip tables with 0 significant DEGs
#   if (nrow(df_sig) == 0) {
#     message(paste("Skipping", nm, "- no significant DEGs"))
#     next
#   }
#   
#   #add filtered table to list
#   toptable_sig_list[[nm]] <- df_sig
# }

#save filtered significant list as an RDS
# saveRDS(toptable_sig_list, file = "data/de/Toptable_sig_list.RDS")

#read in my toptable list for downstream analysis
toptable_list <- readRDS("data/de/Toptable_list.RDS")
toptable_sig_list <- readRDS("data/de/Toptable_sig_list.RDS")

```

```{r Joined Toptable 24hr}
#I want to make a toptable that has all of the data for both species all combined and not just a list

#define the treatments to pull and species
# tx_24 <- c("TUN24", "THA24", "DOX24", "NUTL24", "LPS24", "TNFa24", "BPA24", "PFOA24")

# spec <- c("H", "C")

#make a function to merge each of these toptables into one
# tables_to_merge <- c()
# for (trt in tx_24) {
#   for (sp in spec) {
#     key <- paste0("V.", trt, "_", sp)
#     if (key %in% names(toptable_list_entrez)) {
#       tables_to_merge[[key]] <- toptable_list_entrez[[key]]
#     }
#   }
# }
# combined_24hr <- tables_to_merge[[1]]
# 
# #add suffix to all other tables before merging to keep track of treatment/species
# for (i in 2:length(tables_to_merge)) {
#   df <- tables_to_merge[[i]]
#   
#   #rename columns to include treatment/species
#   df <- df %>%
#     rename_with(~ paste0(names(tables_to_merge)[i], "_", .), -Entrez_ID)
#   
#   #full join by Entrez_ID
#   combined_24hr <- full_join(combined_24hr, df, by = "Entrez_ID")
# }

# saveRDS(combined_24hr, "data/de/Combined_24hr_Toptable_EMP_251001.RDS")

combined_24hr <- readRDS("data/de/Combined_24hr_Toptable_EMP_251001.RDS")

```

```{r Volcano Plots DEGs All Timepoints}
#now I want to create a function to plot volcano plots for all comparisons

generate_volcano_plot <- function(toptable, title) {
  
  #check for entrezid 
  if(!"Entrez_ID" %in% colnames(toptable)) stop("Entrez_ID col not present")
  
  #make significance labels
  toptable <- toptable %>% 
    mutate(Significance = case_when(
      logFC > 0 & adj.P.Val < 0.05 ~ "Upregulated",
      logFC < 0 & adj.P.Val < 0.05 ~ "Downregulated",
      TRUE  ~ "Not Significant"
    ))
  
  #factor significance
  toptable$Significance <- factor(
    toptable$Significance, 
    levels = c("Upregulated", 
               "Not Significant", 
               "Downregulated")
  )
  
  #count genes in each category
  upgenes <- sum(toptable$Significance == "Upregulated")
  nsgenes <- sum(toptable$Significance == "Not Significant")
  downgenes <- sum(toptable$Significance == "Downregulated")
  
  #skip the plot if there are no up and down genes
  if(upgenes == 0 & downgenes == 0) {
    message(paste("Skipping", title, "- no significant DEGs"))
    return(NULL)
  }
  
  #labels for legend
  legend_lab <- c(
    paste0("Upregulated: ", upgenes),
    paste0("Not Significant: ", nsgenes),
    paste0("Downregulated: ", downgenes)
  )
  
  #colors 
  color_map <- c("Upregulated" = "blue",
                 "Not Significant" = "grey30",
                 "Downregulated" = "red")
  
  #generate volcano plots
  p <- ggplot(toptable, aes(x = logFC, 
                            y = -log10(P.Value),
                            color = Significance)) +
    geom_point_rast(alpha = 0.8, size = 3) +
    scale_color_manual(values = color_map, 
                       labels = legend_lab,
                       breaks = c("Upregulated",
                                  "Not Significant", 
                                  "Downregulated"))  +
    xlim(-10,10) +
    labs(title = title, 
         x = expression("log"[2]*"FC"),
         y = expression("-log"[10]*"p-value")) +
    theme_custom() +
    theme(legend.position = "right")
  
  return(p)
}

#now generate a volcano plot for each set
volcano_plots_sig <- lapply(names(toptable_list), function(nm) {
  generate_volcano_plot(toptable_list[[nm]], title = nm)
})
names(volcano_plots_sig) <- names(toptable_list)

#display all
for (plot_name in names(volcano_plots_sig)) {
  print(volcano_plots_sig[[plot_name]])
}

#now save each of these as pdf and png files
# for (plot_name in names(volcano_plots_sig)) {
#   
#   if (is.null(plot_name) || is.na(plot_name) || is.null(volcano_plots_sig[[plot_name]])) {
#     message(paste("skipping", plot_name, "- plot does not have DEGs"))
#     next
#   }
#   
#   p <- volcano_plots_sig[[plot_name]]
#   
#   filename_base <- paste0("Volcano_Plot_", plot_name, "_EMP")
#   
#   save_plot(
#     plot = p,
#     filename = filename_base,
#     folder = output_folder
#   )
# }

```

#Add SYMBOL column
```{r Add HGNC Symbol to each toptable}
#in order to plot the logFC heatmaps, I want to add a SYMBOL column to each of my toptables based on their Entrez_ID

#make a loop to do this since it might take a while
# toptable_list <- lapply(toptable_list, function(tt) {
#   tt <- tt %>% 
#     mutate(
#       Entrez_ID = rownames(tt),
#       SYMBOL = mapIds(org.Hs.eg.db, 
#                       keys = Entrez_ID,
#                       column = "SYMBOL",
#                       keytype = "ENTREZID",
#                       multiVals = "first")
#     )
#   rownames(tt) <- NULL
#   
#   return(tt)
# })

#resave your toptable list so you can keep the symbol col
# saveRDS(toptable_list, "data/de/Toptable_list_entrez.RDS")

#read in your new toptable_list
toptable_list <- readRDS("data/de/Toptable_list_entrez.RDS")

```

#DEGs
```{r Specify your DEGs lists}
#loop through and save my toptable csvs into easier names to maneuver for later analysis - ie Toptable_V.TUN2_C into TUN2_C

#first save all of the toptables that now have entrez and symbol as csv in my DEGs folder

# output <- "data/de/DEGs"
# 
# #save all data frames in toptable_list to csv
# walk2(toptable_list, names(toptable_list), ~ {
#   file_path <- file.path(output, paste0("Toptable_", .y, ".csv"))
#   write.csv(.x, file_path, row.names = FALSE)
#   message("Saved: ", file_path)
# })

#all of the csvs are saved

#now create a load_deg function to load them as needed from this path
load_deg <- function(name, folder = "data/de/DEGs") {
  file <- file.path(folder, paste0("Toptable_V.", name, ".csv"))
  read.csv(file)
}

```

```{r Define gene sets for later analysis}
#define your gene sets DIC and DDR

#here is the set with categories, symbol, and entrezid
fonoudi_genes <- readRDS("data/genesets/Fonoudi_gene_set.RDS")
ddr_genes <- readRDS("data/genesets/DDR_gene_set.RDS")
dic_genes <- readRDS("data/genesets/DIC_gene_set.RDS")

#here are the ids where it's only a vector of Entrez_ID
fonoudi_ids <- readRDS("data/genesets/Fonoudi_genes_entrezid.RDS")
ddr_ids <- readRDS("data/genesets/DDR_genes_entrezid.RDS")
dic_ids <- readRDS("data/genesets/DIC_genes_entrezid.RDS")

```

##logFC Heatmaps
```{r logFC HM DDR DIC 24hr}

# toptable_24hr_names <- names(toptable_list)[grepl("24", names(toptable_list))]
# 
# toptable_list_24 <- toptable_list[toptable_24hr_names]

# saveRDS(toptable_list_24, "data/de/Toptable_list_24hr.RDS")

#read in my toptable list for 24hr
toptable_list_24 <- readRDS("data/de/Toptable_list_24hr.RDS")

#read in my gene sets
ddr_genes
fonoudi_genes

#read in colors
spec_col <- readRDS("data/theme/species_category_color_palette.RDS")
stim_col <- readRDS("data/theme/stimulus_color_palette_all.RDS")
ddr_col <- c("DOX" = "#3B4DA3", "NUTL" = "#A0CFF8")
ind_col <- readRDS("data/theme/individual_category_color_palette.RDS")

#make my ddr heatmap


```

#DDR logFC HM
```{r logFC HM DDR genes 24hr DOX NUTL}
# Load 24hr DEGs
DOX_24_H <- read.csv("data/de/Toptable_V.DOX24_H.csv")
DOX_24_C <- read.csv("data/de/Toptable_V.DOX24_C.csv")
NUTL_24_H <- read.csv("data/de/Toptable_V.NUTL24_H.csv")
NUTL_24_C <- read.csv("data/de/Toptable_V.NUTL24_C.csv")

#I named them without the time since I don't want that annotated onto the heatmaps

# DDR gene set (with categories and Entrez IDs)
ddr_genes <- readRDS("data/genesets/DDR_gene_set.RDS")

ddr_genes <- ddr_genes %>% 
  mutate(Entrez_ID = as.character(Entrez_ID))

ddr_ids <- readRDS("data/genesets/DDR_genes_entrezid.RDS") %>% 
  as.character()

# Metadata table (columns: Final_sample_name, Stimulus, Species, Individual)
metadata_24 <- metadata %>% 
  dplyr::filter(Time == "24")

#extract relevant DEG values
extract_data_DDR <- function(df, name) {
  df %>%
    filter(Entrez_ID %in% ddr_ids) %>%
    mutate(
      Gene = mapIds(org.Hs.eg.db, 
                    as.character(Entrez_ID),
                    column = "SYMBOL", 
                    keytype = "ENTREZID", 
                    multiVals = "first"),
      Condition = name,
      Signif = ifelse(adj.P.Val < 0.05, "*", "")
    )
}

deg_list_ddr24 <- list("DOX_H" = DOX_24_H, 
                       "NUTL_H" = NUTL_24_H, 
                       "DOX_C" = DOX_24_C,
                       "NUTL_C" = NUTL_24_C
)

deg_list_ddr24 <- lapply(deg_list_ddr24, function(df) {
  df %>% mutate(Entrez_ID = as.character(Entrez_ID))
})

all_data_ddr <- bind_rows(mapply(extract_data_DDR, deg_list_ddr24, names(deg_list_ddr24), SIMPLIFY = FALSE)) %>% 
  left_join(ddr_genes, by = "Entrez_ID")

#create my matrices
logFC_matddr <- acast(all_data_ddr, Gene ~ Condition, value.var = "logFC")
signif_matddr <- acast(all_data_ddr, Gene ~ Condition, value.var = "Signif")

#set desired order
desired_order <- c("DOX_H", "DOX_C", "NUTL_H", "NUTL_C")

logFC_mat_ddr <- logFC_matddr[, desired_order, drop = FALSE]
signif_mat_ddr <- signif_matddr[, desired_order, drop = FALSE]

#make column annotations (stimuli species)
meta_ddr <- str_split_fixed(colnames(logFC_mat_ddr), "_", 2)

col_annot <- HeatmapAnnotation(
  Stimulus = meta_ddr[, 1],
  Species = meta_ddr[, 2],
  col = list(
    Stimulus = ddr_col,
    Species = spec_col
  ),
  annotation_height = unit(c(1, 1, 1), "cm")
)

#row annotation (categories)
gene_order_df_ddr <- all_data_ddr %>% 
  distinct(Gene, Category) %>% 
  arrange(factor(Category, 
                 levels = sort(unique(ddr_genes$Category))), Gene)

ordered_genes_ddr <- gene_order_df_ddr$Gene
logFC_mat_ddr <- logFC_mat_ddr[ordered_genes_ddr, ]
signif_mat_ddr <- signif_mat_ddr[ordered_genes_ddr, ]

category_levels <- sort(unique(ddr_genes$Category))

category_cols_ddr <- setNames(
  c("darkorange", "steelblue", "darkgreen", "firebrick",
    "orchid", "mediumpurple"),
  sort(unique(ddr_genes$Category))
)

ha_left_ddr <- rowAnnotation(
  Category = gene_order_df_ddr$Category,
  col = list(Category = category_cols_ddr),
  annotation_name_side = "top"
)

#plot the heatmap
DDR_logFC_HM <- Heatmap(logFC_mat_ddr,
        name = "logFC",
        top_annotation = col_annot,
        left_annotation = ha_left_ddr,
        cluster_columns = FALSE,
        cluster_rows = FALSE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        row_names_gp = gpar(fontsize = 10),
        column_title = "DDR Genes logFC HM",
        column_title_gp = gpar(fontsize = 12),
        cell_fun = function(j, i, x, y, width, height, fill) {
          #draw a black border
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
          #add significance for DEGs
          grid.text(signif_mat_ddr[i,j], x, y, gp = gpar(fontsize = 10))
        })

#save this as a pdf and png
#make a function specifically to save ComplexHeatmaps since save_plot doesn't work with these

save_complex_heatmap <- function(ht, filename, folder = ".", width = 6, height = 10, dpi = 300, add_date = TRUE) {
  if (missing(filename)) stop("Please provide a filename (without extension) for the plot.")
  
  date_str <- if (add_date) paste0("_", format(Sys.Date(), "%y%m%d")) else ""
  
  pdf_file <- file.path(folder, paste0(filename, date_str, ".pdf"))
  png_file <- file.path(folder, paste0(filename, date_str, ".png"))
  
  # Save PDF
  cairo_pdf(pdf_file, width = width, height = height)
  draw(ht)
  dev.off()
  message("Saved PDF: ", pdf_file)
  
  # Save PNG
  png(png_file, width = width, height = height, units = "in", res = dpi)
  draw(ht)
  dev.off()
  message("Saved PNG: ", png_file)
}

print(DDR_logFC_HM)

#now save my heatmap
# save_complex_heatmap(DDR_logFC_HM, 
#                      "DDR_logFC_HM_EMP", 
#                      folder = output_folder)


```


#DOX Cardiotox logFC HM
```{r Dox Cardiotox Fonoudi genes logFC HM}
# Load 24hr DEGs
DOX_24_H <- read.csv("data/de/Toptable_V.DOX24_H.csv")
DOX_24_C <- read.csv("data/de/Toptable_V.DOX24_C.csv")
NUTL_24_H <- read.csv("data/de/Toptable_V.NUTL24_H.csv")
NUTL_24_C <- read.csv("data/de/Toptable_V.NUTL24_C.csv")

#I named them without the time since I don't want that annotated onto the heatmaps

# DDR gene set (with categories and Entrez IDs)
fnd_genes <- readRDS("data/genesets/Fonoudi_gene_set.RDS")

fnd_genes <- fnd_genes %>% 
  mutate(Entrez_ID = as.character(Entrez_ID))

fnd_ids <- readRDS("data/genesets/Fonoudi_genes_entrezid.RDS") %>% 
  as.character()

# Metadata table (columns: Final_sample_name, Stimulus, Species, Individual)
metadata_24 <- metadata %>% 
  dplyr::filter(Time == "24")

#extract relevant DEG values
extract_data_fnd <- function(df, name) {
  df %>%
    filter(Entrez_ID %in% fnd_ids) %>%
    mutate(
      Gene = mapIds(org.Hs.eg.db, 
                    as.character(Entrez_ID),
                    column = "SYMBOL", 
                    keytype = "ENTREZID", 
                    multiVals = "first"),
      Condition = name,
      Signif = ifelse(adj.P.Val < 0.05, "*", "")
    )
}

deg_list_fnd24 <- list("DOX_H" = DOX_24_H, 
                       "NUTL_H" = NUTL_24_H, 
                       "DOX_C" = DOX_24_C,
                       "NUTL_C" = NUTL_24_C
)

deg_list_fnd24 <- lapply(deg_list_fnd24, function(df) {
  df %>% mutate(Entrez_ID = as.character(Entrez_ID))
})

all_data_fnd <- bind_rows(mapply(extract_data_fnd, deg_list_fnd24, names(deg_list_fnd24), SIMPLIFY = FALSE)) %>% 
  left_join(fnd_genes, by = "Entrez_ID")

#create my matrices
logFC_matfnd <- acast(all_data_fnd, Gene ~ Condition, value.var = "logFC")
signif_matfnd <- acast(all_data_fnd, Gene ~ Condition, value.var = "Signif")

#set desired order
desired_order <- c("DOX_H", "DOX_C", "NUTL_H", "NUTL_C")

logFC_mat_fnd <- logFC_matfnd[, desired_order, drop = FALSE]
signif_mat_fnd <- signif_matfnd[, desired_order, drop = FALSE]

#make column annotations (stimuli species)
meta_fnd <- str_split_fixed(colnames(logFC_mat_fnd), "_", 2)

col_annot <- HeatmapAnnotation(
  Stimulus = meta_fnd[, 1],
  Species = meta_fnd[, 2],
  col = list(
    Stimulus = ddr_col,
    Species = spec_col
  ),
  annotation_height = unit(c(1, 1, 1), "cm")
)

#row annotation (categories)
gene_order_df_fnd <- all_data_fnd %>% 
  distinct(Gene, Category) %>% 
  arrange(factor(Category, 
                 levels = sort(unique(fnd_genes$Category))), Gene)

ordered_genes_fnd <- gene_order_df_fnd$Gene
logFC_mat_fnd <- logFC_mat_fnd[ordered_genes_fnd, ]
signif_mat_fnd <- signif_mat_fnd[ordered_genes_fnd, ]

category_levels <- sort(unique(fnd_genes$Category))

category_cols_fnd <- setNames(
  c("darkorange", "steelblue", "darkgreen", "firebrick",
    "orchid", "mediumpurple",  "gold"),
  sort(unique(fnd_genes$Category))
)

ha_left_fnd <- rowAnnotation(
  Category = gene_order_df_fnd$Category,
  col = list(Category = category_cols_fnd),
  annotation_name_side = "top"
)

#plot the heatmap
FND_logFC_HM <- Heatmap(logFC_mat_fnd,
        name = "logFC",
        top_annotation = col_annot,
        left_annotation = ha_left_fnd,
        cluster_columns = FALSE,
        cluster_rows = FALSE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        row_names_gp = gpar(fontsize = 10),
        column_title = "Fonoudi Genes logFC HM",
        column_title_gp = gpar(fontsize = 12),
        cell_fun = function(j, i, x, y, width, height, fill) {
          #draw a black border
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
          #add significance for DEGs
          grid.text(signif_mat_fnd[i,j], x, y, gp = gpar(fontsize = 10))
        })

print(FND_logFC_HM)

#now save my heatmap
# save_complex_heatmap(FND_logFC_HM, 
#                      "FND_logFC_HM_EMP", 
#                      folder = output_folder)


```


```{r DIC logFC HM}
#read in list of genes in DIC gene set (from Matthews et al., 2021)
dic_genes <- readRDS("data/genesets/DIC_genes_set.RDS")
dic_ids <- readRDS("data/genesets/DIC_genes_entrezid.RDS")

# Load 24hr DEGs
DOX_24_H <- read.csv("data/de/Toptable_V.DOX24_H.csv")
DOX_24_C <- read.csv("data/de/Toptable_V.DOX24_C.csv")
NUTL_24_H <- read.csv("data/de/Toptable_V.NUTL24_H.csv")
NUTL_24_C <- read.csv("data/de/Toptable_V.NUTL24_C.csv")

#I named them without the time since I don't want that annotated onto the heatmaps

# DDR gene set (with categories and Entrez IDs)
dic_genes <- dic_genes %>% 
  mutate(Entrez_ID = as.character(Entrez_ID))

dic_ids <- readRDS("data/genesets/DIC_genes_entrezid.RDS") %>% 
  as.character()

# Metadata table (columns: Final_sample_name, Stimulus, Species, Individual)
metadata_24 <- metadata %>% 
  dplyr::filter(Time == "24")

#extract relevant DEG values
extract_data_dic <- function(df, name) {
  df %>%
    filter(Entrez_ID %in% dic_ids) %>%
    mutate(
      Gene = mapIds(org.Hs.eg.db, 
                    as.character(Entrez_ID),
                    column = "SYMBOL", 
                    keytype = "ENTREZID", 
                    multiVals = "first"),
      Condition = name,
      Signif = ifelse(adj.P.Val < 0.05, "*", "")
    )
}

deg_list_dic24 <- list("DOX_H" = DOX_24_H, 
                       "NUTL_H" = NUTL_24_H, 
                       "DOX_C" = DOX_24_C,
                       "NUTL_C" = NUTL_24_C
)

deg_list_dic24 <- lapply(deg_list_dic24, function(df) {
  df %>% mutate(Entrez_ID = as.character(Entrez_ID))
})

all_data_dic <- bind_rows(mapply(extract_data_dic, deg_list_dic24, names(deg_list_dic24), SIMPLIFY = FALSE)) %>% 
  left_join(dic_genes, by = "Entrez_ID")

#create my matrices
logFC_matdic <- acast(all_data_dic, Gene ~ Condition, value.var = "logFC")
signif_matdic <- acast(all_data_dic, Gene ~ Condition, value.var = "Signif")

#set desired order
desired_order <- c("DOX_H", "DOX_C", "NUTL_H", "NUTL_C")

logFC_mat_dic <- logFC_matdic[, desired_order, drop = FALSE]
signif_mat_dic <- signif_matdic[, desired_order, drop = FALSE]

#make column annotations (stimuli species)
meta_dic <- str_split_fixed(colnames(logFC_mat_dic), "_", 2)

col_annot <- HeatmapAnnotation(
  Stimulus = meta_dic[, 1],
  Species = meta_dic[, 2],
  col = list(
    Stimulus = ddr_col,
    Species = spec_col
  ),
  annotation_height = unit(c(1, 1, 1), "cm")
)

#don't need row annotation for this set, no categories

#plot the heatmap
DIC_logFC_HM <- Heatmap(logFC_mat_dic,
        name = "logFC",
        top_annotation = col_annot,
        cluster_columns = FALSE,
        cluster_rows = TRUE,
        show_row_names = TRUE,
        show_column_names = FALSE,
        row_names_gp = gpar(fontsize = 10),
        column_title = "DIC Genes logFC HM",
        column_title_gp = gpar(fontsize = 12),
        cell_fun = function(j, i, x, y, width, height, fill) {
          #draw a black border
          grid.rect(x = x, y = y, width = width, height = height,
                    gp = gpar(col = "black", lwd = 0.5, fill = fill))
          #add significance for DEGs
          grid.text(signif_mat_dic[i,j], x, y, gp = gpar(fontsize = 10))
        })

print(DIC_logFC_HM)

#now save my heatmap
save_complex_heatmap(DIC_logFC_HM,
                     "DIC_logFC_HM_EMP",
                     folder = output_folder)


```



#Treatment Comparisons
```{r Compare DEGs Venn Diagrams}
#make comparisons across all 24hr pairwise treatment DEGs
#venn diagram 

stimuli <- c("TUN24", "THA24", "DOX24", "NUTL24",
             "LPS24", "TNFa24", "BPA24", "PFOA24")

#loop through the defined treatments
for (st in stimuli) {
  
  #make object names
  nm_H <- paste0("V.", st, "_H")
  nm_C <- paste0("V.", st, "_C")
  
  #skip if not in the toptable list for DEGs
  if (!(nm_H %in% names(toptable_list_24)) & !(nm_C %in% names(toptable_list_24))) {
    message("skipping", st, "- no DEGs H and C")
    next
  }
  
  h_genes <- if (nm_H %in% names(toptable_list_24)) {
    df_h <- toptable_list_24[[nm_H]]
    df_h$Entrez_ID <- as.character(df_h$Entrez_ID)
    df_h$Entrez_ID[!is.na(df_h$adj.P.Val) & df_h$adj.P.Val < 0.05]
  } else character(0)
  
  c_genes <- if (nm_C %in% names(toptable_list_24)) {
    df_c <- toptable_list_24[[nm_C]]
    df_c$Entrez_ID <- as.character(df_c$Entrez_ID)
    df_c$Entrez_ID[!is.na(df_c$adj.P.Val) & df_c$adj.P.Val < 0.05]
  }
  
  # #pull DEGs by filtering for adj.p < 0.05
  deg_H <- toptable_list_24[[nm_H]] %>%
    dplyr::filter(adj.P.Val < 0.05) %>%
    pull(Entrez_ID)
  deg_C <- toptable_list_24[[nm_H]] %>%
    dplyr::filter(adj.P.Val < 0.05) %>%
    pull(Entrez_ID)

  #if both empty - skip
  if (length(h_genes) == 0 & length(c_genes) == 0) {
    message("skipping", st, "- no DEGs H and C")
    next
  }
  
  #create venn list
  venn_list <- list(
    H = h_genes,
    C = c_genes
  )
  
  #fit proportional venn
  fit <- euler(venn_list, shape = "ellipse")
  
  #plot proportional venn
  plot_title <- paste0("DEGs H vs C - ", st)
  p <- plot(fit,
       fills = list(fill = c("#740949", "#A6AA59"), alpha = 0.6),
       labels = TRUE,
       quantities = TRUE, 
       main = plot_title
       )
  
  print(p)
  
  # #save as png and pdf in my output_folder
  # png_filename <- file.path(output_folder,
  #                           paste0("PropVenn_", st, "_EMP_251002.png"))
  # png(png_filename, width = 6, height = 6, units = "in", res = 300)
  # print(
  #   plot(fit,
  #      fills = list(fill = c("#740949", "#A6AA59"), alpha = 0.6),
  #      labels = TRUE,
  #      quantities = TRUE,
  #      main = plot_title
  # )
  # )
  # dev.off()
  # 
  # pdf_filename <- file.path(output_folder,
  #                           paste0("PropVenn_", st, "_EMP_251002.pdf"))
  # cairo_pdf(pdf_filename, width = 6, height = 6)
  # print(
  #   plot(fit,
  #      fills = list(fill = c("#740949", "#A6AA59"), alpha = 0.6),
  #      labels = TRUE,
  #      quantities = TRUE,
  #      main = plot_title
  # )
  # )
  # dev.off()
  # 
  # message("Saved Euler diagram for ", st)

}


```

```{r Scatterplots logFC All Treatments 24hr}
#read in my toptable_list_24
toptable_list_24 <- readRDS("data/de/Toptable_list_24hr.RDS")

toptable_list_24 <- lapply(toptable_list_24, function(df) {
  df %>% mutate(Entrez_ID = as.character(Entrez_ID))
})

#define stimuli
stimuli <- c("TUN24", "THA24", "DOX24", "NUTL24",
             "LPS24", "TNFa24", "BPA24", "PFOA24")

#store correlation data
spearman_results <- data.frame(
  Stimulus = character(),
  Spearman_rho = numeric(),
  pvalue = numeric(),
  stringsAsFactors = FALSE
)


for (st in stimuli) {
  #make object names
  nm_H <- paste0("V.", st, "_H")
  nm_C <- paste0("V.", st, "_C")
  
  #skip if not in the toptable list for DEGs
  if (!(nm_H %in% names(toptable_list_24)) & !(nm_C %in% names(toptable_list_24))) {
    message("skipping", st, "- no H and C found")
    next
  }
  
  #pull my data and default to empty tibble if missing
  df_h <- toptable_list_24[[nm_H]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))
  
  df_c <- toptable_list_24[[nm_C]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))

  
  merged_df <- data.frame(
    Entrez_ID = df_h$Entrez_ID,
    logFC_H = as.numeric(df_h$logFC),
    logFC_C = as.numeric(df_c$logFC)
  )
  
  #calculate spearman correlation and plot rho on plot
  cor_test <- cor.test(merged_df$logFC_H, merged_df$logFC_C, 
                       method = "spearman")
  rho <- round(cor_test$estimate, 3)
  pval <- ifelse(cor_test$p.value < 0.001, "<0.001",
                 round(cor_test$p.value, 3))
  cor_label <- paste0("rho = ", rho, "\n", "p = ", pval)
  
  spearman_results <- rbind(spearman_results,
                            data.frame(Stimulus = st,
                                       Rho = rho,
                                       pvalue = cor_test$p.value))
  
  stim_name <- sub("24.*", "", st)
  point_color <- ifelse(stim_name %in% names(stim_col),
                        stim_col[stim_name], "gray50")
  
  #plot scatterplots
  #use fixed coordinates for same visual scale
  p <- ggplot(merged_df, aes(x = logFC_H, y = logFC_C)) +
    rasterize(geom_point(color = point_color), dpi = 300) +
    geom_abline(slope = 1, intercept = 0, 
                linetype = "dashed", color = "black", lwd = 1) +
    coord_fixed(xlim = c(-9, 9),
                ylim = c(-9, 9)) +
    labs(title = paste0("logFC H vs C - ", st),
         x = expression(log[2]*"FC H"),
         y = expression(log[2]*"FC C")
    ) +
    annotate("text", x = -8.5, y = 8.5, label = cor_label,
             hjust = 0, vjust = 1, size = 4) +
    theme_custom()
  
  print(p)
  
  # Save using your save_plot function
  # filename_base <- paste0("logFC_scatter_", st, "_EMP")
  # save_plot(plot = p,
  #           filename = filename_base,
  #           folder = output_folder)
  # 
  # message("Saved scatterplot for ", st)
  # 
}

print(spearman_results)

```


#Species Specific Genes
```{r Identify Species-Specific genes from logFC}
#I want to take my dataset and identify whether genes are responding in one species, both species, or neither species

#read in toptable list at 24hr and filter down to DOX and NUTL only
# toptable_list_24

toptable_list_ddr_24 <- toptable_list_24[grepl("DOX|NUTL", names(toptable_list_24))]

#pull out the factors I'm interested in for now
ddr_stim <- c("DOX", "NUTL")
species <- c("H", "C")

#read in my two gene sets that I want to assess the response in across species from the heatmaps above
gene_sets_ddr <- list(
  ddr = ddr_ids,
  dtox = fonoudi_ids
)

#now make a function to extract logFC and adj.P.Val for each
extract_species_df <- function(stim, sp) {
  #make object names
  nm <- paste0("V.", stim, "24_", sp)
  
  if(!nm %in% names(toptable_list_ddr_24)) {
    stop(paste("Table not found: ", nm))
  }

  df <- toptable_list_ddr_24[[nm]] %>% 
    dplyr::select(Entrez_ID, logFC, adj.P.Val) %>% 
    mutate(
      Entrez_ID = as.character(Entrez_ID),
      logFC = as.character(logFC),
      adj.P.Val = as.numeric(adj.P.Val)
    )
  
  #add a suffix for species and stimulus
  colnames(df)[2:3] <- paste0(colnames(df)[2:3], "_", sp)
  return(df)
}

#now extract your four datasets
DOX_H <- extract_species_df("DOX", "H")
DOX_C <- extract_species_df("DOX", "C")
NUTL_H <- extract_species_df("NUTL", "H")
NUTL_C <- extract_species_df("NUTL", "C")

#merge datasets by Entrez_ID
merged_ddr <- DOX_H %>% 
  left_join(DOX_C, by = "Entrez_ID") %>% 
  left_join(NUTL_H, by = "Entrez_ID") %>% 
  left_join(NUTL_C, by = "Entrez_ID")


#add tags for degs based on sig adj.P.Val < 0.05
merged_ddr <- merged_ddr %>% 
  mutate(
    sig_H_DOX = adj.P.Val_H.x < 0.05, #add .x from merge
    sig_C_DOX = adj.P.Val_C.x < 0.05,
    sig_H_NUTL = adj.P.Val_H.y < 0.05, #add .y from merge
    sig_C_NUTL = adj.P.Val_C.y < 0.05
  )


#add response direction
merged_ddr <- merged_ddr %>% 
  mutate(
    Dir_H_DOX = case_when(
      sig_H_DOX & logFC_H.x > 0 ~ "Up",
      sig_H_DOX & logFC_H.x < 0 ~ "Down",
      TRUE ~ "NR"
    ),
    Dir_C_DOX = case_when(
      sig_C_DOX & logFC_C.x > 0 ~ "Up",
      sig_C_DOX & logFC_C.x < 0 ~ "Down",
      TRUE ~ "NR"
    ),
    Dir_H_NUTL = case_when(
      sig_H_NUTL & logFC_H.y > 0 ~ "Up",
      sig_H_NUTL & logFC_H.y < 0 ~ "Down",
      TRUE ~ "NR"
    ),
    Dir_C_NUTL = case_when(
      sig_C_NUTL & logFC_C.y > 0 ~ "Up",
      sig_C_NUTL & logFC_C.y < 0 ~ "Down",
      TRUE ~ "NR"
    )
  )

#classify response patterns between species to find species-specific deg
classify_response <- function(df, col_H, col_C) {
  case_when(
    df[[col_H]] == "NR" & df[[col_C]] == "NR" ~ "NR",
    df[[col_H]] != "NR" & df[[col_C]] != "NR" & 
      df [[col_H]] == df[[col_C]] ~ "Conserved",
    df[[col_H]] != "NR" & df[[col_C]] != "NR" &
      df[[col_H]] != df[[col_C]] ~ "Divergent",
    df[[col_H]] != "NR" & df[[col_C]] == "NR" ~ "H_specific",
    df[[col_H]] == "NR" & df[[col_C]] != "NR" ~ "C_specific",
    TRUE ~ "Other"
  )
}

merged_ddr <- merged_ddr %>% 
  mutate(
    Response_DOX = classify_response(., 
                                     "Dir_H_DOX",
                                     "Dir_C_DOX"),
    Response_NUTL = classify_response(., 
                                      "Dir_H_NUTL",
                                      "Dir_C_NUTL")
  )

#make my merged table and save as csv
# write.csv(merged_ddr, "data/de/DDR_DTOX_24hr_Species_Responses_AllGenes_EMP_251003.csv",
#           row.names = FALSE)

#get the column SYMBOL in there since it's easy to map it to there
# identical(merged_ddr$Entrez_ID, toptable_list_24$V.TUN24_H$Entrez_ID)
#TRUE

# merged_ddr$SYMBOL <- toptable_list_24$V.TUN24_H$SYMBOL

#now subset your data to find only those that are DE
full_ddr <- merged_ddr

deg_ddr_24 <- full_ddr %>% 
  dplyr::filter(
    sig_H_DOX | sig_C_DOX | sig_H_NUTL | sig_C_NUTL
  )

# saveRDS(deg_resp_ddr, "DDR_DEGs_direction.RDS")

```



```{r Species Specific logFC Scatterplots}

#define stimuli
ddr_stim <- c("DOX", "NUTL")

spearman_res <- data.frame(
  Stimulus = character(),
  Rho = numeric(),
  pvalue = numeric(),
  stringsAsFactors = FALSE
)

plots <- list()

for (st in ddr_stim) {
  # Make object names
  nm_H <- paste0("V.", st, "24_H")
  nm_C <- paste0("V.", st, "24_C")
  
  if (!(nm_H %in% names(toptable_list_24)) | !(nm_C %in% names(toptable_list_24))) {
    message("Skipping ", st, " (missing H or C)")
    next
  }
  
  df_h <- toptable_list_24[[nm_H]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))
  df_c <- toptable_list_24[[nm_C]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))
  
  # Merge by Entrez ID
  merged_df <- data.frame(
    Entrez_ID = df_h$Entrez_ID,
    logFC_H = as.numeric(df_h$logFC),
    logFC_C = as.numeric(df_c$logFC)
  )
  
  # Identify opposite-direction genes
  merged_df <- merged_df %>%
    mutate(OppositeDir = case_when(
      logFC_H > 0 & logFC_C < 0 ~ "Opposite",
      logFC_H < 0 & logFC_C > 0 ~ "Opposite",
      TRUE ~ "Same"
    ))
  
  n_opposite <- sum(merged_df$OppositeDir == "Opposite")
  
  # Spearman correlation
  cor_test <- cor.test(merged_df$logFC_H, merged_df$logFC_C, method = "spearman")
  rho <- round(cor_test$estimate, 2)
  pval <- ifelse(cor_test$p.value < 0.001, "<0.001", round(cor_test$p.value, 3))
  cor_label <- paste0("rho = ", rho, "\n", "p = ", pval)
  
  spearman_res <- rbind(spearman_res,
                            data.frame(Stimulus = st,
                                       Rho = rho,
                                       pvalue = cor_test$p.value))
  
  stim_name <- gsub("24$", "", st)
  stim_color <- ifelse(stim_name %in% names(stim_col), stim_col[stim_name], "purple")
  
  # Scatterplot
  p <- ggplot(merged_df, aes(x = logFC_H, y = logFC_C)) +
    rasterise(geom_point(data = merged_df %>% filter(OppositeDir == "Same"),
                         color = "black", alpha = 0.5, size = 0.8), dpi = 300) +
    rasterise(geom_point(data = merged_df %>% filter(OppositeDir == "Opposite"),
                         color = stim_color, alpha = 0.9, size = 1.1), dpi = 300) +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray30") +
    geom_hline(yintercept = 0, linetype = "solid", color = "black") +
    geom_vline(xintercept = 0, linetype = "solid", color = "black") +
    coord_fixed(xlim = c(-5, 5), ylim = c(-5, 5)) +
    labs(title = paste0("logFC H vs C - ", stim_name),
         x = expression(log[2]*"FC H"),
         y = expression(log[2]*"FC C")) +
    annotate("text", x = -4.5, y = 4.5, label = cor_label,
             hjust = 0, vjust = 1, size = 4) +
    annotate("text", x = 4.5, y = -4.5, 
             label = paste0("n = ", n_opposite),
             hjust = 1, vjust = 0, size = 4, color = stim_color) +
    theme_custom()
  
  plots[[st]] <- p
  print(p)
}

print(spearman_res)


combined_plot <- plots[["DOX"]] + plots[["NUTL"]] +
  plot_annotation(theme = theme_custom())

# Save with your custom save_plot function
# save_plot(combined_plot, 
#           "DOX_NUTL_logFC_correlation_EMP", 
#           width = 10, height = 5,
#           folder = output_folder)


```

#Species Specific logFC Correlation Scatterplots - all stimuli
```{r Species Specific logFC Cor Scatter all}
#define stimuli
# stimuli <- c("TUN", "THA", "DOX", "NUTL",
#              "LPS", "TNFa", "BPA", "PFOA")
# saveRDS(stimuli, "data/theme/stimuli_abbrv_list.RDS")

stimuli <- readRDS("data/theme/stimuli_abbrv_list.RDS")

spearman_res_all <- data.frame(
  Stimulus = character(),
  Rho = numeric(),
  pvalue = numeric(),
  stringsAsFactors = FALSE
)

plots_all <- list()

for (st in stimuli) {
  # Make object names
  nm_H <- paste0("V.", st, "24_H")
  nm_C <- paste0("V.", st, "24_C")
  
  if (!(nm_H %in% names(toptable_list_24)) | !(nm_C %in% names(toptable_list_24))) {
    message("Skipping ", st, " (missing H or C)")
    next
  }
  
  df_h <- toptable_list_24[[nm_H]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))
  df_c <- toptable_list_24[[nm_C]] %>% 
    mutate(Entrez_ID = as.character(Entrez_ID))
  
  # Merge by Entrez ID
  merged_df <- data.frame(
    Entrez_ID = df_h$Entrez_ID,
    logFC_H = as.numeric(df_h$logFC),
    logFC_C = as.numeric(df_c$logFC)
  )
  
  # Identify opposite-direction genes
  merged_df <- merged_df %>%
    mutate(OppositeDir = case_when(
      logFC_H > 0 & logFC_C < 0 ~ "Opposite",
      logFC_H < 0 & logFC_C > 0 ~ "Opposite",
      TRUE ~ "Same"
    ))
  
  n_opposite <- sum(merged_df$OppositeDir == "Opposite")
  
  # Spearman correlation
  cor_test <- cor.test(merged_df$logFC_H, merged_df$logFC_C, 
                       method = "spearman")
  rho <- round(cor_test$estimate, 2)
  pval <- ifelse(cor_test$p.value < 0.001, "<0.001",
                 round(cor_test$p.value, 3))
  cor_label <- paste0("rho = ", rho, "\n", "p = ", pval)
  
  spearman_res_all <- rbind(spearman_res_all,
                            data.frame(Stimulus = st,
                                       Rho = rho,
                                       pvalue = cor_test$p.value))
  
  stim_name <- gsub("24$", "", st)
  stim_color <- ifelse(stim_name %in% names(stim_col), stim_col[stim_name], "black")
  
  # Scatterplot
  p <- ggplot(merged_df, aes(x = logFC_H, y = logFC_C)) +
    rasterise(geom_point(data = merged_df %>% filter(OppositeDir == "Same"),
                         color = "black", alpha = 0.5, size = 0.8), 
              dpi = 300) +
    rasterise(geom_point(data = merged_df %>% 
                           filter(OppositeDir == "Opposite"),
                         color = stim_color, alpha = 0.9, 
                         size = 1.1), dpi = 300) +
    geom_abline(slope = 1, intercept = 0, 
                linetype = "dashed", color = "gray30") +
    geom_hline(yintercept = 0, linetype = "solid", color = "black") +
    geom_vline(xintercept = 0, linetype = "solid", color = "black") +
    coord_fixed(xlim = c(-5, 5), ylim = c(-5, 5)) +
    labs(title = paste0("logFC H vs C - ", stim_name),
         x = expression(log[2]*"FC H"),
         y = expression(log[2]*"FC C")) +
    annotate("text", x = -4.5, y = 4.5, label = cor_label,
             hjust = 0, vjust = 1, size = 4) +
    annotate("text", x = 4.5, y = -4.5, 
             label = paste0("n = ", n_opposite),
             hjust = 1, vjust = 0, size = 4, color = stim_color) +
    theme_custom()
  
  plots_all[[st]] <- p
  print(p)
}

print(spearman_res_all)

#print single plots
# for (st in stimuli) {
#   p <- plots_all[[st]]
#   print(p)
#   filename_base <- paste0("Species_logFC_Cor_scatter_", st, "_EMP")
#   save_plot(plot = p, filename = filename_base, folder = output_folder, width = 5, height = 5)
# }

#save as 2x4 grid
# combined_plot_2x4 <- wrap_plots(
#   plots_all[["TUN"]],
#   plots_all[["THA"]],
#   plots_all[["DOX"]],
#   plots_all[["NUTL"]],
#   plots_all[["LPS"]],
#   plots_all[["TNFa"]],
#   plots_all[["BPA"]],
#   plots_all[["PFOA"]],
#   ncol = 4,
#   nrow = 2
# ) + plot_annotation(theme = theme_custom())

#save â€” each plot as 5x5 â†’ total 20x10 inches
# save_plot(
#   plot = combined_plot_2x4,
#   filename = "Species_logFC_Cor_scatter_allstimuli_2x4_EMP",
#   folder = output_folder,
#   width = 20,
#   height = 10
# )

#now save this as a 4x2 grid just in case
# combined_plot_4x2 <- wrap_plots(
#   plots_all[["TUN"]],
#   plots_all[["THA"]],
#   plots_all[["DOX"]],
#   plots_all[["NUTL"]],
#   plots_all[["LPS"]],
#   plots_all[["TNFa"]],
#   plots_all[["BPA"]],
#   plots_all[["PFOA"]],
#   ncol = 2,
#   nrow = 4
# ) + plot_annotation(theme = theme_custom())

#save â€” each plot 5x5 = total 10x20 inches
# save_plot(
#   plot = combined_plot_4x2,
#   filename = "Species_logFC_Cor_scatter_allstimuli_4x2_EMP",
#   folder = output_folder,
#   width = 10,
#   height = 20
# )


```

#Pull out Species Specific Genes
```{r Species Specific Genes List}
#I want to pull out genes that are specific to each species to plot examples later
#I can use my merged_df from above with all stimuli to make a dataframe for classification of these genes based on logFC

#Human Specific = logFC H > 0, logFC C < 0
#Chimp Specific = logFC H < 0, logFC C > 0
#Shared = logFC H </> 0, logFC C </> 0


# merged_df_all <- merged_df %>% 
#   mutate(
#     Category = case_when(
#       logFC_H > 0 & logFC_C < 0 ~ "Human-specific",
#       logFC_H < 0 & logFC_C > 0 ~ "Chimp-specific",
#       sign(logFC_H) == sign(logFC_C) ~ "Shared"
#     )
#   ) %>% 
#   dplyr::select(Entrez_ID, logFC_H, logFC_C, Category)

# saveRDS(merged_df_all, "data/de/species_specific_classification_df.RDS")

#read in your new df 
merged_df_all <- readRDS("data/de/species_specific_classification_df.RDS")

#make one that is classified by stimulus as well
# all_gene_classified <- do.call(rbind, lapply(stimuli, function(st) {
#   nm_H <- paste0("V.", st, "24_H")
#   nm_C <- paste0("V.", st, "24_C")
# 
#   if (!(nm_H %in% names(toptable_list_24)) | !(nm_C %in% names(toptable_list_24))) {
#     return(NULL)
#   }
#   
#   df_h <- toptable_list_24[[nm_H]] %>% 
#     mutate(
#       Entrez_ID = as.character(Entrez_ID),
#       SYMBOL = as.character(SYMBOL),
#       adj.P.Val_H = adj.P.Val
#     )
# 
#   df_c <- toptable_list_24[[nm_C]] %>%
#     mutate(Entrez_ID = as.character(Entrez_ID),
#            SYMBOL = as.character(SYMBOL),
#            adj.P.Val_C = adj.P.Val
#     )
# 
#   df_species_spec <- data.frame(
#     Entrez_ID = df_h$Entrez_ID,
#     SYMBOL = df_h$SYMBOL,
#     logFC_H = as.numeric(df_h$logFC),
#     logFC_C = as.numeric(df_c$logFC),
#     adj.P.Val_H = as.numeric(df_h$adj.P.Val_H),
#     adj.P.Val_C = as.numeric(df_c$adj.P.Val_C)
#   ) %>%
#     mutate(
#       Category = case_when(
#         logFC_H > 0 & logFC_C < 0 ~ "Human-specific",
#         logFC_H < 0 & logFC_C > 0 ~ "Chimp-specific",
#         sign(logFC_H) == sign(logFC_C) ~ "Shared"
#       ),
#       Stimulus = st,
#       Time = "24"
#     )
# 
#   return(df_species_spec)
# }))

# saveRDS(all_gene_classified, "data/de/specspec/all_classified_genes_df.RDS")

#now pull the data to get the genes for each category
# h_spec_genes <- all_gene_classified %>% filter(Category == "Human-specific")
# saveRDS(h_spec_genes, "data/de/specspec/human_specific_df.RDS")
# h_spec_genes_list <- h_spec_genes$Entrez_ID
# saveRDS(h_spec_genes_list, "data/de/specspec/human_specific_entrezid_list.RDS")

# c_spec_genes <- all_gene_classified %>% filter(Category == "Chimp-specific")
# saveRDS(c_spec_genes, "data/de/specspec/chimp_specific_df.RDS")
# c_spec_genes_list <- c_spec_genes$Entrez_ID
# saveRDS(c_spec_genes_list, "data/de/specspec/chimp_specific_entrezid_list.RDS")

# hc_shared_genes <- all_gene_classified %>% filter(Category == "Shared")
# saveRDS(hc_shared_genes, "data/de/specspec/shared_genes_df_hc.RDS")
# hc_shared_genes_list <- hc_shared_genes$Entrez_ID
# saveRDS(hc_shared_genes_list, "data/de/specspec/shared_hc_genes_entrezid_list.RDS")

#keep this for posterity, but I have saved each of these as RDS objects
all_gene_classified <- readRDS("data/de/specspec/all_classified_genes_df.RDS")

#make a short table
all_gene_classified_table <- table(all_gene_classified$Stimulus, all_gene_classified$Category)

print(all_gene_classified_table)

#create a list of stimulus specific and species specific dataframes
# species_gene_list <- list()
# 
# for (st in stimuli) {
#   nm_H <- paste0("V.", st, "24_H")
#   nm_C <- paste0("V.", st, "24_C")
#   
#   df_h <- toptable_list_24[[nm_H]] %>% 
#     dplyr::select(SYMBOL, Entrez_ID, adj.P.Val, logFC)
#   df_c <- toptable_list_24[[nm_C]] %>% 
#     dplyr::select(SYMBOL, Entrez_ID, adj.P.Val, logFC)
#     
#     df_species_spec <- data.frame(
#     Entrez_ID = df_h$Entrez_ID,
#     SYMBOL = df_h$SYMBOL,
#     logFC_H = as.numeric(df_h$logFC),
#     logFC_C = as.numeric(df_c$logFC)
#   ) %>%
#     mutate(
#       Category = case_when(
#         logFC_H > 0 & logFC_C < 0 ~ "Human-specific",
#         logFC_H < 0 & logFC_C > 0 ~ "Chimp-specific",
#         sign(logFC_H) == sign(logFC_C) ~ "Shared"
#       ),
#       Stimulus = st,
#     )
#   
#   # save the dataframe to the list
#   species_gene_list[[st]] <- df_species_spec
# }

# saveRDS(species_gene_list, "data/de/specspec/species_categorized_genes_bystim_listofdf.RDS")

species_gene_list <- readRDS("data/de/specspec/species_categorized_genes_bystim_listofdf.RDS")

dic_symbols <- dic_genes$SYMBOL

#check if genes are differentially expressed
de_specspec_dic <- all_gene_classified %>%
  filter(SYMBOL %in% dic_symbols,      # must be in your DDR gene list
         (Category == "Human-specific" & adj.P.Val_H < 0.05) |
         (Category == "Chimp-specific" & adj.P.Val_C < 0.05) |
         (Category == "Shared" & (adj.P.Val_H < 0.05 | adj.P.Val_C < 0.05))) %>%
  arrange(Category)

de_specspec_all <- all_gene_classified %>%
  filter((Category == "Human-specific" & adj.P.Val_H < 0.05) |
         (Category == "Chimp-specific" & adj.P.Val_C < 0.05) |
         (Category == "Shared" & (adj.P.Val_H < 0.05 | adj.P.Val_C < 0.05))) %>%
  arrange(Category)

#now I have filtered for DE genes within these two sets across categories

```

```{r Plot log2cpm example genes across species}
#plot a human specific gene, chimp specific gene, and a shared gene
#look first at those that are in your dic_ids list to see if any are present across stimuli and then plot

#read in my filtered cpm matrix
hc_cpm_mat_24 <- readRDS("data/counts/hc_filtered_cpm_matrix_24hr.RDS")
colnames(hc_cpm_mat_24) <- metadata_24$Final_sample_name
#change the column names to match my metadata

#filter all_gene_classified to only have DOX and NUTL
gene_classified_ddr <- de_specspec_all %>% 
  dplyr::filter(Stimulus %in% c("DOX", "NUTL")) %>% 
  arrange(Category)

dic_gene_classified <- gene_classified_ddr %>% 
  dplyr::filter(SYMBOL %in% dic_symbols) %>% 
  arrange(Category)

category_counts_ddr <- dic_gene_classified %>% 
  group_by(Category) %>% 
  summarise(n_genes = n_distinct(SYMBOL))

#return all of the genes that are species specific
human_DOX <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "DOX") %>% 
  filter(Category == "Human-specific") %>% 
  pull(SYMBOL)

chimp_DOX <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "DOX") %>% 
  filter(Category == "Chimp-specific") %>% 
  pull(SYMBOL)

shared_DOX <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "DOX") %>% 
  filter(Category == "Shared") %>% 
  pull(SYMBOL)

#now the nutl groups
human_NUTL <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "NUTL") %>% 
  filter(Category == "Human-specific") %>% 
  pull(SYMBOL)

chimp_NUTL <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "NUTL") %>% 
  filter(Category == "Chimp-specific") %>% 
  pull(SYMBOL)

shared_NUTL <- gene_classified_ddr %>% 
  dplyr::filter(Stimulus == "NUTL") %>% 
  filter(Category == "Shared") %>% 
  pull(SYMBOL)

order_bps <- c("DOX_H", "NUTL_H", "DMSO_H",
               "DOX_C", "NUTL_C", "DMSO_C")

# combine all groups into a list
gene_groups <- list(
  human_DOX = human_DOX,
  human_NUTL = human_NUTL,
  chimp_DOX = chimp_DOX,
  chimp_NUTL = chimp_NUTL,
  shared_DOX = shared_DOX,
  shared_NUTL = shared_NUTL
)


```


